<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sally Beard" />

<meta name="date" content="2022-09-02" />

<title>FOX Preeclampsia RUVSeq-edgeR Differential Expression Analysis</title>

<script src="site_libs/header-attrs-2.4/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
      </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Fetal-oxygenation-study</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sabeard4/Fetal-oxygenation-study">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">FOX Preeclampsia RUVSeq-edgeR Differential Expression Analysis</h1>
<h4 class="author">Sally Beard</h4>
<h4 class="date">2022-09-02</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Fetal-oxygenation-study/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220826code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220826)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220826code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220826)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsabeard4Fetaloxygenationstudytreef3c81c9f286faed8336c8b92f3353e75f1ae38cetargetblankf3c81c9a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sabeard4/Fetal-oxygenation-study/tree/f3c81c9f286faed8336c8b92f3353e75f1ae38ce" target="_blank">f3c81c9</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsabeard4Fetaloxygenationstudytreef3c81c9f286faed8336c8b92f3353e75f1ae38cetargetblankf3c81c9a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sabeard4/Fetal-oxygenation-study/tree/f3c81c9f286faed8336c8b92f3353e75f1ae38ce" target="_blank">f3c81c9</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  analysis/FOX PE RNA-seq DGE.Rmd
    Untracked:  data/HK_genes.txt
    Untracked:  data/counts/
    Untracked:  data/sample_metadata.csv
    Untracked:  figure/

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/FOX_PE_RUVSeq.Rmd</code>) and HTML (<code>docs/FOX_PE_RUVSeq.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sabeard4/Fetal-oxygenation-study/blob/f3c81c9f286faed8336c8b92f3353e75f1ae38ce/analysis/FOX_PE_RUVSeq.Rmd" target="_blank">f3c81c9</a>
</td>
<td>
Sally Beard
</td>
<td>
2022-09-07
</td>
<td>
Update file input path and add chunk names for images
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sabeard4/Fetal-oxygenation-study/bbd7972d299f031debfc69bf9c66885a5273a5ff/docs/FOX_PE_RUVSeq.html" target="_blank">bbd7972</a>
</td>
<td>
Sally Beard
</td>
<td>
2022-09-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sabeard4/Fetal-oxygenation-study/blob/6fe2c91d958342e8ec2b911918a7f8fd6131c481/analysis/FOX_PE_RUVSeq.Rmd" target="_blank">6fe2c91</a>
</td>
<td>
Sally Beard
</td>
<td>
2022-09-06
</td>
<td>
FOX PE RUVSeq-edgeR DGE analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>The sequence data used in the following analysis was generated at AGRF for the <a href="https://link.springer.com/article/10.1186/s12916-020-01605-x">FOX study</a>. RNA was isolated from whole blood PAXgene tubes and Illumina’s TruSeq stranded RNA sample preparation kit with globin depletion was used to prepare libraries for sequencing. Libraries were pooled and clustered using the Illumina cBot system, followed by sequencing on the Illumina HiSeq 2000 platform to obtain 50 base pair single-end reads. Each library was split over two lanes, which generated two sequence files per sample. Fastq files were aligned against the human genome GRCh38 using <code>hisat2</code>, then files were sorted with <code>SAMtools</code>, before reads aligning to exons were counted using <code>htseq</code> with default parameters. Technical duplicates run in different sequencing lanes overlapped on MDS plots, indicating there was no batch effect due to lane, therefore in this analysis counts from technical duplicates were summed for further analysis.</p>
<p>Only samples collected as cases in the FOX study were included in the following analysis. These consisted of maternal blood from women with preterm growth-restricted fetuses, recruited from six sites across Australia and New Zealand. Samples from 115 women were included. Patient characteristics are shown in table 1, followed by the analysis output. The two groups considered are those with preeclampsia according to ISSHP <a href="https://www.sciencedirect.com/science/article/abs/pii/S2210778918301260?via%3Dihub">guidelines</a> and those without.</p>
<pre class="r"><code>#import required libraries
library(limma)
library(Glimma)
library(edgeR)
library(tidyverse)
library(EDASeq)
library(RColorBrewer)
library(tableone)
library(knitr)
library(kableExtra)
library(Homo.sapiens)
library(patchwork)
library(NMF)
library(here)
library(RUVSeq)
library(ggrepel)</code></pre>
<div id="data-import" class="section level2">
<h2>Data import</h2>
<p>Import sample information and create table of characteristics.</p>
<pre class="r"><code>metadata &lt;- read.csv(&quot;data/sample_metadata.csv&quot;)
#create a variable list
vars &lt;- c(&quot;age&quot;, &quot;bmi&quot;, &quot;diabetes&quot;, &quot;hypertension&quot;,&quot;gestation_sampling&quot;, &quot;gestation_delivery&quot;, &quot;birthweight&quot;, &quot;birthweight_centile&quot;, &quot;fetal_sex&quot;, &quot;cord_blood_pH&quot;)
vars2 &lt;- list(age = &quot;Maternal age&quot;,
             bmi = &quot;Maternal BMI&quot;,
             diabetes = &quot;Diabetes&quot;,
             hypertension = &quot;Preeclampsia&quot;,
             gestation_sampling = &quot;Gestation at sample collection (weeks)&quot;,
             gestation_delivery = &quot;Gestation at delivery (weeks)&quot;,
             birthweight = &quot;Fetal birthweight (g)&quot;,
             birthweight_centile = &quot;Birthweight centile&quot;,
             fetal_sex = &quot;Fetal sex&quot;,
             cord_blood_pH = &quot;Cord blood pH&quot;)
labelled::var_label(metadata) &lt;- vars2

#create Table 1 stratified by hypertension
metadata$hypertension &lt;- relevel(metadata$hypertension, &quot;None&quot;)
tableOne &lt;- CreateTableOne(vars = vars, strata = &quot;hypertension&quot;, data = metadata, testNonNormal = kruskal.test)
kable_styling(kableone(tableOne, caption=&quot;Table 1: Patient characteristics for FOX cases included in RNA-seq analysis&quot;, nonnormal = c(&quot;birthweight_centile&quot;, &quot;cord_blood_pH&quot;, &quot;bmi&quot;), varLabels = TRUE, dropEqual = TRUE))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Table 1: Patient characteristics for FOX cases included in RNA-seq analysis
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
None
</th>
<th style="text-align:left;">
Preeclampsia
</th>
<th style="text-align:left;">
p
</th>
<th style="text-align:left;">
test
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
n
</td>
<td style="text-align:left;">
44
</td>
<td style="text-align:left;">
71
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Maternal age (mean (SD))
</td>
<td style="text-align:left;">
31.45 (5.58)
</td>
<td style="text-align:left;">
32.32 (5.71)
</td>
<td style="text-align:left;">
0.425
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Maternal BMI (median [IQR])
</td>
<td style="text-align:left;">
24.00 [21.00, 27.00]
</td>
<td style="text-align:left;">
27.00 [22.00, 32.00]
</td>
<td style="text-align:left;">
0.024
</td>
<td style="text-align:left;">
nonnorm
</td>
</tr>
<tr>
<td style="text-align:left;">
Diabetes (%)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
0.405
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
GDM
</td>
<td style="text-align:left;">
5 (11.4)
</td>
<td style="text-align:left;">
5 ( 7.0)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
37 (84.1)
</td>
<td style="text-align:left;">
63 ( 88.7)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
T1DM
</td>
<td style="text-align:left;">
0 ( 0.0)
</td>
<td style="text-align:left;">
2 ( 2.8)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
T2DM
</td>
<td style="text-align:left;">
2 ( 4.5)
</td>
<td style="text-align:left;">
1 ( 1.4)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
hypertension (%)
</td>
<td style="text-align:left;">
0 ( 0.0)
</td>
<td style="text-align:left;">
71 (100.0)
</td>
<td style="text-align:left;">
&lt;0.001
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Gestation at sample collection (weeks) (mean (SD))
</td>
<td style="text-align:left;">
31.03 (1.87)
</td>
<td style="text-align:left;">
29.94 (2.34)
</td>
<td style="text-align:left;">
0.010
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Gestation at delivery (weeks) (mean (SD))
</td>
<td style="text-align:left;">
31.03 (1.87)
</td>
<td style="text-align:left;">
29.94 (2.34)
</td>
<td style="text-align:left;">
0.010
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Fetal birthweight (g) (mean (SD))
</td>
<td style="text-align:left;">
1086.27 (352.60)
</td>
<td style="text-align:left;">
1050.18 (333.41)
</td>
<td style="text-align:left;">
0.582
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Birthweight centile (median [IQR])
</td>
<td style="text-align:left;">
3.05 [1.08, 6.60]
</td>
<td style="text-align:left;">
8.40 [2.55, 16.45]
</td>
<td style="text-align:left;">
0.005
</td>
<td style="text-align:left;">
nonnorm
</td>
</tr>
<tr>
<td style="text-align:left;">
Fetal sex (%)
</td>
<td style="text-align:left;">
27 (61.4)
</td>
<td style="text-align:left;">
41 ( 57.7)
</td>
<td style="text-align:left;">
0.851
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Cord blood pH (median [IQR])
</td>
<td style="text-align:left;">
7.28 [7.25, 7.31]
</td>
<td style="text-align:left;">
7.27 [7.22, 7.30]
</td>
<td style="text-align:left;">
0.330
</td>
<td style="text-align:left;">
nonnorm
</td>
</tr>
</tbody>
</table>
<p>Set up <code>DGElist</code> object for downstream analysis and add sample-specific information.</p>
<pre class="r"><code>#list count files from working directory
files = list.files(&quot;/Users/beards/rnaseq_FOX/FOX_case_counts_tidy_PEvNoPE&quot;, pattern = &quot;[counts]$&quot;)

#edgeR has a function readDGE for combining separate count files into one matrix of counts in one step
counts &lt;- readDGE(files, path = &quot;/Users/beards/rnaseq_FOX/FOX_case_counts_tidy_PEvNoPE&quot;, columns = c(1,3), header = FALSE)

#remove meta tags as library size includes counts from the meta tags
MetaTags &lt;- grep(&quot;^__&quot;, rownames(counts))
counts &lt;- counts[-MetaTags, ]

#update library size to reflect mapped reads
counts$samples$lib.size &lt;- colSums(counts$counts)

#obtain sample names from file names
sample &lt;- strsplit2(files, &quot;_&quot;)[,1]
counts$samples$sample &lt;- as.factor(sample)
samplenames &lt;- counts$samples$sample

#sum counts from technical replicates
counts &lt;- sumTechReps(counts, ID=samplenames)

#Add sample-level information related to experimental design (and possible confounding factors) to the count matrix
counts$samples$site &lt;- as.factor(metadata$site)
counts$samples$sample &lt;- as.factor(metadata$sample)
counts$samples$sex &lt;- as.factor(metadata$fetal_sex)
counts$samples$group &lt;- as.factor(metadata$hypertension)
counts$samples$diabetes &lt;- as.factor(metadata$diabetes)
counts$samples$steroids &lt;- as.factor(metadata$steroids)

#import information for each gene
geneid &lt;- rownames(counts)
genes &lt;- select(Homo.sapiens, keys=geneid, columns=c(&quot;SYMBOL&quot;, &quot;TXCHROM&quot;, &quot;ENTREZID&quot;), 
                keytype=&quot;ENSEMBL&quot;)
genes &lt;- genes[!duplicated(genes$ENSEMBL),]
counts$genes &lt;- genes

#remove genes without ENTREZ IDs
keep &lt;- !is.na(counts$genes$ENTREZID) &amp; !is.null(counts$genes$ENTREZID)
counts &lt;- counts[keep, ]

#generate library statistics
cpm &lt;- cpm(counts)
lcpm &lt;- cpm(counts, log=TRUE)
L &lt;- mean(counts$samples$lib.size) * 1e-6
M &lt;- median(counts$samples$lib.size) * 1e-6</code></pre>
</div>
<div id="quality-control" class="section level2">
<h2>Quality Control</h2>
<p>Genes with no or low counts are unlikely to be biologically important, they provide little evidence for differential expression, and interfere with statistical approximations, so they were filtered out before performing the analysis. Genes with low or no counts (counts per million reads that corresponded to less than ~10 reads in any 44 samples (smallest group size)) were filtered out of the analysis, based on the recommendations in the edgeR documentation. Both smaller (&gt;2) and larger (&lt;100) thresholds were tested, lower thresholds detected more significant genes on average, higher thresholds detected less differentially expressed genes, but those that were detected are likely to be more robust. The threshold chosen for the analysis is commonly used in the literature and provided a good compromise.</p>
<pre class="r"><code>#filter out genes with low expression
keep.exprs &lt;- filterByExpr(counts, group=counts$samples$group)
counts_FOX &lt;- counts[keep.exprs,, keep.lib.sizes=FALSE]
samplenames &lt;- colnames(counts_FOX)</code></pre>
<p>Plotting the distribution of log-CPM values (below) showed that before filtering (A) many genes in each sample had low or no expression, with negative log-CPM values. These genes were removed from the analysis after filtering (B). Filtering didn’t improve the uniformity between samples however.</p>
<pre class="r"><code>#make density plots for counts before and after filtering out genes with low expression
par(mfrow = c(1,1))
lcpmz &lt;- lcpm
lcpm.cutoff &lt;- log2(10/M + 2/L)
nsamples &lt;- ncol(counts)
col &lt;- scales::hue_pal()(nsamples)
plot(density(lcpm[,1]), col=col[1], lwd=2, ylim=c(0,0.4), las=2, main=&quot;&quot;, xlab=&quot;&quot;)
  title(main=&quot;A. Raw data&quot;, xlab=&quot;Log-cpm&quot;)
abline(v=lcpm.cutoff, lty=3)
for (i in 2:nsamples){
  den &lt;- density(lcpm[,i])
  lines(den$x, den$y, col=col[i], lwd=2)
}</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/density-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>lcpm2 &lt;- cpm(counts_FOX, log=TRUE)
plot(density(lcpm2[,1]), col=col[1], lwd=2, ylim=c(0,0.4), las=2, main=&quot;&quot;, xlab=&quot;&quot;)
title(main=&quot;B. Filtered data&quot;, xlab=&quot;Log-cpm&quot;)
abline(v=lcpm.cutoff, lty=3)
for (i in 2:nsamples){
  den &lt;- density(lcpm2[,i])
  lines(den$x, den$y, col=col[i], lwd=2)
}</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/density-plot-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>We suspected that effective library size might have been contributing to the lack of uniformity between samples in the density plots. Colouring the samples by effective library size showed that the distribution of counts was similar for samples with larger libraries (8-12 million reads) where two peaks were seen after filtering, one at logCPM ~-1.5 and one at 5 (B). Samples with smaller libraries tended to have a single peak between 0 and 3 log-CPM (B).</p>
<pre class="r"><code>#create binds for effective samples size
lib.size &lt;- counts_FOX$samples$lib.size
#set up cut-off values
breaks &lt;- c(0,2000000,4000000,6000000,8000000,10000000,12000000)
#specify bin labels
tags &lt;- c(&quot;[0-2m)&quot;, &quot;[2-4m)&quot;, &quot;[4-6m)&quot;, &quot;[6-8m)&quot;, &quot;[8-10m)&quot;, &quot;[10-12m)&quot;)
#bucket library sizes into bins
group_tags &lt;- cut(lib.size,
                  breaks=breaks,
                  include.lowest=TRUE,
                  right=FALSE,
                  labels=tags)
#add new tags to count matrix
counts_FOX$samples$lib_size &lt;- as.factor(group_tags)

#make density plots for counts before and after filtering out genes with low expression
par(mfrow = c(1,1))
lcpmz &lt;- lcpm
lcpm.cutoff &lt;- log2(10/M + 2/L)
nsamples &lt;- ncol(counts)
col.lib &lt;- counts_FOX$samples$lib_size
levels(col.lib) &lt;- brewer.pal(nlevels(col.lib), &quot;Set1&quot;)
plot(density(lcpmz[,1]), col = as.character(col.lib[1]), lwd = 1, ylim = c(0, 0.4), las = 2, 
     main = &quot;&quot;, xlab = &quot;&quot;)
title(main = &quot;A. Raw data&quot;, xlab = &quot;Log-cpm&quot;)
abline(v = lcpm.cutoff, lty = 3)
for (i in 2:nsamples){
  den &lt;- density(lcpmz[,i])
  lines(den$x, den$y, col = as.character(col.lib[i]), lwd = 1)
}
legend(&quot;topright&quot;,legend=levels(counts_FOX$samples$lib_size), col=levels(col.lib), pch=16, title =&quot;Library size (millions)&quot;)</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/density-plot-coloured-lib.size-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>lcpmy &lt;- cpm(counts_FOX, log=TRUE)
plot(density(lcpmy[,1]), col = as.character(col.lib[1]), lwd = 1, ylim = c(0, 0.4), las = 2, 
     main = &quot;&quot;, xlab = &quot;&quot;)
title(main = &quot;B. Filtered data&quot;, xlab = &quot;Log-cpm&quot;)
abline(v = lcpm.cutoff, lty = 3)
for (i in 2:nsamples){
  den &lt;- density(lcpmy[,i])
  lines(den$x, den$y, col = as.character(col.lib[i]), lwd = 1)
}
legend(&quot;topright&quot;,legend=levels(counts_FOX$samples$lib_size), col=levels(col.lib), pch=16, title =&quot;Library size (millions)&quot; )</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/density-plot-coloured-lib.size-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>TMM normalisation was applied to account for any highly expressed genes monopolising the sequencing and to account for the underlying differences in the distributions of expressed transcripts between samples.</p>
<pre class="r"><code>#apply TMM normalisation
norm_FOX &lt;- calcNormFactors(counts_FOX, method = &quot;TMM&quot;)</code></pre>
<p>Although ~30 million reads per sample were obtained from the sequencing, effective library sizes varied from around 2 million reads to more than 10 million reads, with the median just under 7 million reads (A). This suggests we are only likely to capture the most abundant maternal RNAs. The range and distribution of expression values look fairly uniform between samples, which is what we expect, and TMM normalisation further improves this.</p>
<pre class="r"><code>dat &lt;- data.frame(lib = norm_FOX$samples$lib.size,
                  status = norm_FOX$samples$group,
                  sample = colnames(norm_FOX))
p1 &lt;- ggplot(dat, aes(x = sample, y = lib, fill = status)) +
  geom_bar(stat = &quot;identity&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = &quot;Sample&quot;, y = &quot;Library size&quot;,
       fill = &quot;Preeclampsia status&quot;, title = &quot;A. Library size after filtering&quot;) +
  geom_hline(yintercept = median(dat$lib), linetype = &quot;dashed&quot;) +
  scale_x_discrete(limits = dat$sample)

dat &lt;- reshape2::melt(cpm(norm_FOX, normalized.lib.sizes = FALSE, log = TRUE),
                      value.name = &quot;cpm&quot;)
dat$status &lt;- rep(norm_FOX$samples$group, each = nrow(norm_FOX))
colnames(dat)[2] &lt;- &quot;sample&quot;
p2 &lt;- ggplot(dat, aes(x = sample, y = cpm, fill = status)) +
  geom_boxplot(show.legend = FALSE, outlier.size = 0.75) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7)) +
  labs(x = &quot;Sample&quot;, y = &quot;Log-CPM&quot;,
       fill = &quot;Preeclampsia status&quot;, title = &quot;B. Expression values after filtering&quot;) +
  geom_hline(yintercept = median(dat$lib), linetype = &quot;dashed&quot;)

dat &lt;- reshape2::melt(cpm(norm_FOX, normalized.lib.sizes = TRUE, log = TRUE),
                      value.name = &quot;cpm&quot;)

dat$status &lt;- rep(norm_FOX$samples$group, each = nrow(norm_FOX))
colnames(dat)[2] &lt;- &quot;sample&quot;
p3 &lt;- ggplot(dat, aes(x = sample, y = cpm, fill = status)) +
  geom_boxplot(show.legend = FALSE, outlier.size = 0.75) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7)) +
  labs(x = &quot;Sample&quot;, y = &quot;Log-CPM&quot;,
       fill = &quot;Preeclampsia status&quot;, title = &quot;C. Expression values after normalisation&quot;) +
  geom_hline(yintercept = median(dat$lib), linetype = &quot;dashed&quot;)

p1</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/expression-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p2</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/expression-plot-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p3</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/expression-plot-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Multi-dimensional scaling (MDS) plots show the largest sources of variation in the data. They are a good way of identifying structure in the data and exploring relationships between samples. The following plots examine the first four principle components coloured for known features of the data including preeclampsia status, fetal sex, collection site, diabates status, steroids and library size. The samples do not cluster by known features of the data, except library size. The lack of clustering by preeclampsia status in any of the first four principle components suggests we might not find any differentially expressed genes. The MDS plots suggest that the largest source of variation in the data is library size, as there is a strong trend towards increasingly smaller library sizes from left to right across the first dimension.</p>
<pre class="r"><code>dims &lt;- list(c(1,2), c(1,3), c(2,3), c(3,4))
vars &lt;- c(&quot;group&quot;, &quot;sex&quot;, &quot;site&quot;, &quot;diabetes&quot;, &quot;steroids&quot;, &quot;lib_size&quot;)
patches &lt;- vector(&quot;list&quot;, length(vars))

for(i in 1:length(vars)){
  p &lt;- vector(&quot;list&quot;, length(dims))
  
  for(j in 1:length(dims)){
    mds &lt;- plotMDS(cpm(norm_FOX, log = TRUE), top = 1000, gene.selection=&quot;common&quot;, 
                   plot = FALSE, dim.plot = dims[[j]])
    dat &lt;- tibble::tibble(x = mds$x, y = mds$y,
                          sample = samplenames,
                          variable = pull(norm_FOX$samples, vars[i]))
    
    p[[j]] &lt;- ggplot(dat, aes(x = x, y = y, colour = variable)) +
      geom_text(aes(label = sample), size = 2.5) +
      labs(x = glue::glue(&quot;Principal component {dims[[j]][1]}&quot;), 
           y = glue::glue(&quot;Principal component {dims[[j]][2]}&quot;),
           colour = vars[i])
  }
  
  patches[[i]] &lt;- wrap_elements(wrap_plots(p, ncol = 2, guides = &quot;collect&quot;) +
    plot_annotation(title = glue::glue(&quot;Coloured by: {vars[i]}&quot;)) &amp;
    theme(legend.position = &quot;bottom&quot;))
  
}

patches</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/mds-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/mds-plot-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/mds-plot-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/mds-plot-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/mds-plot-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[6]]</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/mds-plot-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ruvseq-analysis" class="section level2">
<h2>RUVseq analysis</h2>
<p>Given that the samples largely overlapped on the MDS plots, and there is a lot of variability that is not clearly associated with any of the known features (except library size), we will use <code>RUVSeq</code> to remove some of the unwanted variation from the data. As per the <code>RUVSeq</code> vignette we will use a set of housekeeping genes, in this case we will estimate unwanted variation using the expression of housekeeping genes defined using the Human BodyMap 2.0 <a href="https://www.tau.ac.il/~elieis/HKG/">data</a> in our dataset.</p>
<p>The heatmap below shows the expression of these housekeeping genes in our samples. In general expression looks reasonably similar, but there are some differences between samples.</p>
<pre class="r"><code>hkGenes &lt;- read.delim(here(&quot;data/HK_genes.txt&quot;), stringsAsFactors = FALSE, 
                      header = FALSE, col.names = c(&quot;SYMBOL&quot;,&quot;REFSEQ&quot;), 
                      strip.white = TRUE)

aheatmap(cpm(norm_FOX[norm_FOX$genes$SYMBOL %in% hkGenes$SYMBOL,], log = TRUE),
         main = &quot;log2 CPM Expression of Housekeeping Genes&quot;, labRow = NA,  
         annCol = list(Preeclampsia = norm_FOX$samples$group, 
                       Diabetes = norm_FOX$samples$diabetes, 
                       Site = norm_FOX$samples$site,
                       Sex = norm_FOX$samples$sex,
                       Lib_size = norm_FOX$samples$lib_size),
         annColors = list(scales::hue_pal()(length(unique(norm_FOX$samples$group))),
                          scales::hue_pal()(length(unique(norm_FOX$samples$diabetes))),
                          scales::hue_pal()(length(unique(norm_FOX$samples$site))),
                          scales::brewer_pal(palette = &quot;Set1&quot;)(length(unique(norm_FOX$samples$sex))),
                          scales::brewer_pal(palette = &quot;Dark2&quot;)(length(unique(norm_FOX$samples$lib_size)))),
         scale = &quot;none&quot;,
         fontsize = 7)</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/HKG-expression-heatmap-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The scree plot shows that most of the variation within the negative control housekeeping genes is captured in the first principle component.</p>
<pre class="r"><code>PCs &lt;- prcomp(t(norm_FOX$counts[which(norm_FOX$genes$SYMBOL %in% hkGenes$SYMBOL),]), 
              center = TRUE, scale = TRUE, retx=TRUE)
loadings = PCs$x # pc loadings
plot(PCs, type=&quot;lines&quot;) # scree plot</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/scree-plot-HKG-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Using the RUVg function, we will estimate the unwanted variation in our data using the HKG for k = 1, 2 and 3.</p>
<pre class="r"><code>yRg1 &lt;- RUVg(norm_FOX$counts, which(norm_FOX$genes$SYMBOL %in% hkGenes$SYMBOL), k=1)
yRg2 &lt;- RUVg(norm_FOX$counts, which(norm_FOX$genes$SYMBOL %in% hkGenes$SYMBOL), k=2)
yRg3 &lt;- RUVg(norm_FOX$counts, which(norm_FOX$genes$SYMBOL %in% hkGenes$SYMBOL), k=3)</code></pre>
<p>The RLE plots below show that RUVg adjustment is a significant improvement over the raw data with k=1. The improvements with k=2 and k=3 are relatively minor in comparison.</p>
<pre class="r"><code>plotRLE(norm_FOX$counts, ylim=c(-0.5,0.5), las=2, main=&quot;Raw&quot;,
        cex.axis = 0.6)</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/RLE-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotRLE(yRg1$normalizedCounts, ylim=c(-0.5,0.5), las=2, main=&quot;RUVg k=1&quot;, 
        cex.axis = 0.6)</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/RLE-plot-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotRLE(yRg2$normalizedCounts, ylim=c(-0.5,0.5), las=2, main=&quot;RUVg k=2&quot;, 
        cex.axis = 0.6)</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/RLE-plot-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotRLE(yRg3$normalizedCounts, ylim=c(-0.5,0.5), las=2, main=&quot;RUVg k=3&quot;, 
        cex.axis = 0.6)</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/RLE-plot-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>The MDS plots of the RUVg normalised counts below show that the samples now have less variability in the first principal component, but there is still no separation between samples with and without preeclampsia, suggesting that we may not find any differentially expressed genes.</p>
<pre class="r"><code>mds &lt;- plotMDS(cpm(norm_FOX$counts, log = TRUE), top = 1000, gene.selection=&quot;common&quot;, 
               plot = FALSE)
dat &lt;- tibble::tibble(x = mds$x, y = mds$y,
                      sample = norm_FOX$samples$sample,
                      status = norm_FOX$samples$group)

p &lt;- ggplot(dat, aes(x = x, y = y, colour = status)) +
  geom_text(aes(label = sample), size = 2.5) +
  labs(x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;,
       colour = &quot;Preeclampsia status&quot;) +
  ggtitle(&quot;Raw&quot;)

mds &lt;- plotMDS(cpm(yRg1$normalizedCounts, log = TRUE), top = 1000, 
               gene.selection=&quot;common&quot;, plot = FALSE)
dat1 &lt;- tibble::tibble(x = mds$x, y = mds$y,
                      sample = norm_FOX$samples$sample,
                      status = norm_FOX$samples$group)

p1 &lt;- ggplot(dat1, aes(x = x, y = y, colour = status)) +
  geom_text(aes(label = sample), size = 2.5) +
  labs(x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;,
       colour = &quot;Preeclampsia status&quot;) +
  ggtitle(&quot;RUVg k=1&quot;)

mds &lt;- plotMDS(cpm(yRg2$normalizedCounts, log = TRUE), top = 1000, 
               gene.selection=&quot;common&quot;, plot = FALSE)
dat2 &lt;- tibble::tibble(x = mds$x, y = mds$y,
                      sample = norm_FOX$samples$sample,
                      status = norm_FOX$samples$group)

p2 &lt;- ggplot(dat2, aes(x = x, y = y, colour = status)) +
  geom_text(aes(label = sample), size = 2.5) +
  labs(x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;,
       colour = &quot;Preeclampsia status&quot;) +
  ggtitle(&quot;RUVg k=2&quot;)

mds &lt;- plotMDS(cpm(yRg3$normalizedCounts, log = TRUE), top = 1000, 
               gene.selection=&quot;common&quot;, plot = FALSE)
dat3 &lt;- tibble::tibble(x = mds$x, y = mds$y,
                      sample = norm_FOX$samples$sample,
                      status = norm_FOX$samples$group)

p3 &lt;- ggplot(dat3, aes(x = x, y = y, colour = status)) +
  geom_text(aes(label = sample), size = 2.5) +
  labs(x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;,
       colour = &quot;Preeclampsia status&quot;) +
  ggtitle(&quot;RUVg k=3&quot;)

((p | p1) / (p2 | p3)) + 
  plot_layout(guides = &quot;collect&quot;) &amp; 
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/MDS-plot-RUVSeq-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>From the MDS plots below we can see that when samples are coloured for library size, RUVg with k=1 removes the effect of library size in the first principal component.</p>
<pre class="r"><code>mds &lt;- plotMDS(cpm(norm_FOX$counts, log = TRUE), top = 1000, gene.selection=&quot;common&quot;, 
               plot = FALSE)
dat &lt;- tibble::tibble(x = mds$x, y = mds$y,
                      sample = norm_FOX$samples$sample,
                      status = norm_FOX$samples$lib_size)

p &lt;- ggplot(dat, aes(x = x, y = y, colour = status)) +
  geom_text(aes(label = sample), size = 2.5) +
  labs(x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;,
       colour = &quot;Effective library size&quot;) +
  ggtitle(&quot;Raw&quot;)

mds &lt;- plotMDS(cpm(yRg1$normalizedCounts, log = TRUE), top = 1000, 
               gene.selection=&quot;common&quot;, plot = FALSE)
dat1 &lt;- tibble::tibble(x = mds$x, y = mds$y,
                      sample = norm_FOX$samples$sample,
                      status = norm_FOX$samples$lib_size)

p1 &lt;- ggplot(dat1, aes(x = x, y = y, colour = status)) +
  geom_text(aes(label = sample), size = 2.5) +
  labs(x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;,
       colour = &quot;Effective library size&quot;) +
  ggtitle(&quot;RUVg k=1&quot;)

mds &lt;- plotMDS(cpm(yRg2$normalizedCounts, log = TRUE), top = 1000, 
               gene.selection=&quot;common&quot;, plot = FALSE)
dat2 &lt;- tibble::tibble(x = mds$x, y = mds$y,
                      sample = norm_FOX$samples$sample,
                      status = norm_FOX$samples$group)
(p + p1) +
  plot_layout(guides = &quot;collect&quot;) &amp; 
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/MDS-plot-RUVSeq-lib.size-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-expression-analysis" class="section level2">
<h2>Differential expression analysis</h2>
<p>Now we will look for differential expression using the <code>edgeR</code> negative binomial approach outlined in the RUVSeq vignette. This uses a design matrix that includes the covariate of interest (preeclampsia status) and the factors of unwanted variation calculated by <code>RUVg</code>. The samples from preeclamptic pregnancies were compared to those without preeclampsia. A summary of the number of differentially expressed genes (table 2) and their details (table 3) are shown below.</p>
<pre class="r"><code>norm &lt;- yRg1
#set up design matrix
design &lt;- model.matrix(~norm_FOX$samples$group + norm$W)
colnames(design)[2] &lt;- &quot;Preeclampsia&quot;

y &lt;- estimateGLMCommonDisp(norm_FOX, design)
y &lt;- estimateGLMTagwiseDisp(y, design)
fit &lt;- glmFit(y, design)
lrt &lt;- glmLRT(fit, coef = 2)

fitSum &lt;- summary(decideTests(lrt, p.value = 0.05))
kable_styling(kable((fitSum), caption=&quot;Tbale 2: Number of differentially expressed genes&quot;))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Tbale 2: Number of differentially expressed genes
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Preeclampsia
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
NotSig
</td>
<td style="text-align:right;">
13485
</td>
</tr>
<tr>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>top &lt;- topTags(lrt, n = 1)$table
kable_styling(kable((top[,c(1,3,5,6,9)]), caption = &quot;Table 3: DE genes in preeclampsia compared to no hypertension&quot;))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Table 3: DE genes in preeclampsia compared to no hypertension
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
ENSEMBL
</th>
<th style="text-align:left;">
SYMBOL
</th>
<th style="text-align:right;">
logFC
</th>
<th style="text-align:right;">
logCPM
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ENSG00000179833
</td>
<td style="text-align:left;">
ENSG00000179833
</td>
<td style="text-align:left;">
SERTAD2
</td>
<td style="text-align:right;">
0.3683104
</td>
<td style="text-align:right;">
5.57876
</td>
<td style="text-align:right;">
0.0249942
</td>
</tr>
</tbody>
</table>
<p>Volcano plot below shows the statistical significance versus the logFC of genes in the differential expression analysis, with only one gene, SERTAD2 with FDR &lt; 0.05.</p>
<pre class="r"><code>topTags(lrt, n = Inf)$table %&gt;% 
  mutate(sig = ifelse(FDR &lt;= 0.05, &quot;&lt;= 0.05&quot;, &quot;&gt; 0.05&quot;)) -&gt; dat

ggplot(dat, aes(x = logFC, y = -log10(PValue), color = sig)) +
  geom_point(alpha = 0.75) +
  ggrepel::geom_text_repel(data = subset(dat, FDR &lt; 0.05), 
            aes(x = logFC, y = -log10(PValue), 
                label = SYMBOL), 
            size = 2, colour = &quot;black&quot;, max.overlaps = 15) +
  labs(x = expression(~log[2]~&quot;(Fold Change)&quot;), 
       y = expression(~-log[10]~&quot;(P-value)&quot;),
       colour = &quot;FDR&quot;) +
  scale_colour_brewer(palette = &quot;Set1&quot;)</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/volcano-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The following plot shows the expression of the differentially expressed gene for samples collected from women with preeclampsia versus those without. Despite the variability in the data and the lack of separation in the MDS plots, we were still able to detect one differentially expressed gene, although the expression values largely overlap.</p>
<pre class="r"><code>dat &lt;- reshape2::melt(cpm(norm$normalizedCounts, log = TRUE),
                      value.name = &quot;cpm&quot;)
dat$group &lt;- rep(y$samples$group, each = nrow(y))
dat$gene &lt;- rep(y$genes$SYMBOL, ncol(y))
ggplot(data = subset(dat, dat$gene == top$SYMBOL), 
       aes(x = group, y = cpm, colour = group)) +
  geom_jitter(width = 0.25) +
  stat_summary(fun = &quot;mean&quot;, geom = &quot;crossbar&quot;) +
  labs(x = &quot;Status&quot;, y = &quot;log2 Norm. CPM&quot;, colour = &quot;Status&quot;) +
  ggtitle(top$SYMBOL) +
  theme(plot.title = element_text(size = 8),
        plot.subtitle = element_text(size = 7),
        axis.title = element_text(size = 8),
        axis.text.x = element_text(size = 7))</code></pre>
<p><img src="figure/FOX_PE_RUVSeq.Rmd/SERTAD2-logCPM-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="summary" class="section level2">
<h2>Summary</h2>
<p>The results of this study were likely influenced by the small and quite variable effective library size, with only relatively highly expressed genes included in the analysis. RUVSeq normalisation was used to account for this and any other unknown variability in the data. We found only one differentially expressed gene in preeclampsia samples compared to samples without preeclampsia, which was SERTAD2.</p>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggrepel_0.9.1                          
 [2] RUVSeq_1.20.0                          
 [3] here_1.0.1                             
 [4] NMF_0.23.0                             
 [5] cluster_2.1.0                          
 [6] rngtools_1.5                           
 [7] pkgmaker_0.32.2                        
 [8] registry_0.5-1                         
 [9] patchwork_1.1.2                        
[10] Homo.sapiens_1.3.1                     
[11] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[12] org.Hs.eg.db_3.10.0                    
[13] GO.db_3.10.0                           
[14] OrganismDbi_1.28.0                     
[15] GenomicFeatures_1.38.2                 
[16] AnnotationDbi_1.48.0                   
[17] kableExtra_1.3.4                       
[18] knitr_1.30                             
[19] tableone_0.13.0                        
[20] RColorBrewer_1.1-2                     
[21] EDASeq_2.20.0                          
[22] ShortRead_1.44.3                       
[23] GenomicAlignments_1.22.1               
[24] SummarizedExperiment_1.16.1            
[25] DelayedArray_0.12.3                    
[26] matrixStats_0.57.0                     
[27] Rsamtools_2.2.3                        
[28] GenomicRanges_1.38.0                   
[29] GenomeInfoDb_1.22.1                    
[30] Biostrings_2.54.0                      
[31] XVector_0.26.0                         
[32] IRanges_2.20.2                         
[33] S4Vectors_0.24.4                       
[34] BiocParallel_1.20.1                    
[35] Biobase_2.46.0                         
[36] BiocGenerics_0.32.0                    
[37] forcats_0.5.0                          
[38] stringr_1.4.0                          
[39] dplyr_1.0.2                            
[40] purrr_0.3.4                            
[41] readr_1.4.0                            
[42] tidyr_1.1.2                            
[43] tibble_3.0.3                           
[44] ggplot2_3.3.2                          
[45] tidyverse_1.3.0                        
[46] edgeR_3.28.1                           
[47] Glimma_1.14.0                          
[48] limma_3.42.2                           
[49] workflowr_1.7.0                        

loaded via a namespace (and not attached):
  [1] R.utils_2.10.1         tidyselect_1.1.0       RSQLite_2.2.1         
  [4] grid_3.6.1             DESeq_1.38.0           munsell_0.5.0         
  [7] codetools_0.2-16       withr_2.3.0            colorspace_1.4-1      
 [10] highr_0.8              rstudioapi_0.11        labeling_0.3          
 [13] git2r_0.27.1           GenomeInfoDbData_1.2.2 hwriter_1.3.2         
 [16] farver_2.0.3           bit64_4.0.5            rprojroot_2.0.3       
 [19] vctrs_0.3.4            generics_0.0.2         xfun_0.18             
 [22] BiocFileCache_1.10.2   R6_2.4.1               doParallel_1.0.17     
 [25] locfit_1.5-9.4         bitops_1.0-6           assertthat_0.2.1      
 [28] promises_1.1.1         scales_1.1.1           gtable_0.3.0          
 [31] processx_3.7.0         rlang_0.4.7            genefilter_1.68.0     
 [34] systemfonts_1.0.4      splines_3.6.1          rtracklayer_1.46.0    
 [37] broom_0.7.1            BiocManager_1.30.10    yaml_2.2.1            
 [40] reshape2_1.4.4         modelr_0.1.8           backports_1.1.10      
 [43] httpuv_1.5.4           RBGL_1.62.1            tools_3.6.1           
 [46] gridBase_0.4-7         ellipsis_0.3.1         Rcpp_1.0.5            
 [49] plyr_1.8.6             progress_1.2.2         zlibbioc_1.32.0       
 [52] RCurl_1.98-1.2         ps_1.3.4               prettyunits_1.1.1     
 [55] openssl_1.4.3          zoo_1.8-8              haven_2.3.1           
 [58] fs_1.5.0               survey_4.1-1           magrittr_1.5          
 [61] reprex_0.3.0           whisker_0.4            aroma.light_3.16.0    
 [64] hms_0.5.3              evaluate_0.14          xtable_1.8-4          
 [67] XML_3.99-0.3           jpeg_0.1-8.1           readxl_1.3.1          
 [70] compiler_3.6.1         biomaRt_2.42.1         crayon_1.3.4          
 [73] R.oo_1.24.0            htmltools_0.5.0        later_1.1.0.1         
 [76] geneplotter_1.64.0     lubridate_1.7.9        DBI_1.1.0             
 [79] dbplyr_1.4.4           MASS_7.3-53            rappdirs_0.3.1        
 [82] Matrix_1.2-18          cli_2.0.2              mitools_2.4           
 [85] R.methodsS3_1.8.1      pkgconfig_2.0.3        getPass_0.2-2         
 [88] xml2_1.3.2             foreach_1.5.0          svglite_2.1.0         
 [91] annotate_1.64.0        webshot_0.5.2          rvest_0.3.6           
 [94] callr_3.7.2            digest_0.6.25          graph_1.64.0          
 [97] rmarkdown_2.4          cellranger_1.1.0       curl_4.3              
[100] lifecycle_0.2.0        jsonlite_1.7.1         viridisLite_0.3.0     
[103] askpass_1.1            fansi_0.4.1            labelled_2.7.0        
[106] pillar_1.4.6           lattice_0.20-41        httr_1.4.2            
[109] survival_3.2-7         glue_1.4.2             png_0.1-7             
[112] iterators_1.0.12       bit_4.0.4              class_7.3-17          
[115] stringi_1.5.3          blob_1.2.1             latticeExtra_0.6-29   
[118] memoise_1.1.0          e1071_1.7-3           </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggrepel_0.9.1                          
 [2] RUVSeq_1.20.0                          
 [3] here_1.0.1                             
 [4] NMF_0.23.0                             
 [5] cluster_2.1.0                          
 [6] rngtools_1.5                           
 [7] pkgmaker_0.32.2                        
 [8] registry_0.5-1                         
 [9] patchwork_1.1.2                        
[10] Homo.sapiens_1.3.1                     
[11] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[12] org.Hs.eg.db_3.10.0                    
[13] GO.db_3.10.0                           
[14] OrganismDbi_1.28.0                     
[15] GenomicFeatures_1.38.2                 
[16] AnnotationDbi_1.48.0                   
[17] kableExtra_1.3.4                       
[18] knitr_1.30                             
[19] tableone_0.13.0                        
[20] RColorBrewer_1.1-2                     
[21] EDASeq_2.20.0                          
[22] ShortRead_1.44.3                       
[23] GenomicAlignments_1.22.1               
[24] SummarizedExperiment_1.16.1            
[25] DelayedArray_0.12.3                    
[26] matrixStats_0.57.0                     
[27] Rsamtools_2.2.3                        
[28] GenomicRanges_1.38.0                   
[29] GenomeInfoDb_1.22.1                    
[30] Biostrings_2.54.0                      
[31] XVector_0.26.0                         
[32] IRanges_2.20.2                         
[33] S4Vectors_0.24.4                       
[34] BiocParallel_1.20.1                    
[35] Biobase_2.46.0                         
[36] BiocGenerics_0.32.0                    
[37] forcats_0.5.0                          
[38] stringr_1.4.0                          
[39] dplyr_1.0.2                            
[40] purrr_0.3.4                            
[41] readr_1.4.0                            
[42] tidyr_1.1.2                            
[43] tibble_3.0.3                           
[44] ggplot2_3.3.2                          
[45] tidyverse_1.3.0                        
[46] edgeR_3.28.1                           
[47] Glimma_1.14.0                          
[48] limma_3.42.2                           
[49] workflowr_1.7.0                        

loaded via a namespace (and not attached):
  [1] R.utils_2.10.1         tidyselect_1.1.0       RSQLite_2.2.1         
  [4] grid_3.6.1             DESeq_1.38.0           munsell_0.5.0         
  [7] codetools_0.2-16       withr_2.3.0            colorspace_1.4-1      
 [10] highr_0.8              rstudioapi_0.11        labeling_0.3          
 [13] git2r_0.27.1           GenomeInfoDbData_1.2.2 hwriter_1.3.2         
 [16] farver_2.0.3           bit64_4.0.5            rprojroot_2.0.3       
 [19] vctrs_0.3.4            generics_0.0.2         xfun_0.18             
 [22] BiocFileCache_1.10.2   R6_2.4.1               doParallel_1.0.17     
 [25] locfit_1.5-9.4         bitops_1.0-6           assertthat_0.2.1      
 [28] promises_1.1.1         scales_1.1.1           gtable_0.3.0          
 [31] processx_3.7.0         rlang_0.4.7            genefilter_1.68.0     
 [34] systemfonts_1.0.4      splines_3.6.1          rtracklayer_1.46.0    
 [37] broom_0.7.1            BiocManager_1.30.10    yaml_2.2.1            
 [40] reshape2_1.4.4         modelr_0.1.8           backports_1.1.10      
 [43] httpuv_1.5.4           RBGL_1.62.1            tools_3.6.1           
 [46] gridBase_0.4-7         ellipsis_0.3.1         Rcpp_1.0.5            
 [49] plyr_1.8.6             progress_1.2.2         zlibbioc_1.32.0       
 [52] RCurl_1.98-1.2         ps_1.3.4               prettyunits_1.1.1     
 [55] openssl_1.4.3          zoo_1.8-8              haven_2.3.1           
 [58] fs_1.5.0               survey_4.1-1           magrittr_1.5          
 [61] reprex_0.3.0           whisker_0.4            aroma.light_3.16.0    
 [64] hms_0.5.3              evaluate_0.14          xtable_1.8-4          
 [67] XML_3.99-0.3           jpeg_0.1-8.1           readxl_1.3.1          
 [70] compiler_3.6.1         biomaRt_2.42.1         crayon_1.3.4          
 [73] R.oo_1.24.0            htmltools_0.5.0        later_1.1.0.1         
 [76] geneplotter_1.64.0     lubridate_1.7.9        DBI_1.1.0             
 [79] dbplyr_1.4.4           MASS_7.3-53            rappdirs_0.3.1        
 [82] Matrix_1.2-18          cli_2.0.2              mitools_2.4           
 [85] R.methodsS3_1.8.1      pkgconfig_2.0.3        getPass_0.2-2         
 [88] xml2_1.3.2             foreach_1.5.0          svglite_2.1.0         
 [91] annotate_1.64.0        webshot_0.5.2          rvest_0.3.6           
 [94] callr_3.7.2            digest_0.6.25          graph_1.64.0          
 [97] rmarkdown_2.4          cellranger_1.1.0       curl_4.3              
[100] lifecycle_0.2.0        jsonlite_1.7.1         viridisLite_0.3.0     
[103] askpass_1.1            fansi_0.4.1            labelled_2.7.0        
[106] pillar_1.4.6           lattice_0.20-41        httr_1.4.2            
[109] survival_3.2-7         glue_1.4.2             png_0.1-7             
[112] iterators_1.0.12       bit_4.0.4              class_7.3-17          
[115] stringi_1.5.3          blob_1.2.1             latticeExtra_0.6-29   
[118] memoise_1.1.0          e1071_1.7-3           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
