<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sally Beard" />

<meta name="date" content="2022-09-27" />

<title>FOX Case-Control limma-voom Differential Expression Analysis</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Fetal-oxygenation-study</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sabeard4/Fetal-oxygenation-study">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">FOX Case-Control limma-voom Differential Expression Analysis</h1>
<h4 class="author">Sally Beard</h4>
<h4 class="date">2022-09-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Fetal-oxygenation-study/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220826code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220826)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220826code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220826)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsabeard4Fetaloxygenationstudytreeb4f08497019cf8c9634776160bfd699462eafa69targetblankb4f0849a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sabeard4/Fetal-oxygenation-study/tree/b4f08497019cf8c9634776160bfd699462eafa69" target="_blank">b4f0849</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsabeard4Fetaloxygenationstudytreeb4f08497019cf8c9634776160bfd699462eafa69targetblankb4f0849a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sabeard4/Fetal-oxygenation-study/tree/b4f08497019cf8c9634776160bfd699462eafa69" target="_blank">b4f0849</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  data/FOX_REDCap_metadata.csv
    Untracked:  data/HK_genes.txt
    Untracked:  data/MSigDB/
    Untracked:  data/sample_metadata.csv
    Untracked:  output/camera_GO.csv
    Untracked:  output/camera_c2.csv
    Untracked:  output/camera_hallmark.csv
    Untracked:  output/goana_all.csv
    Untracked:  output/goana_treat.csv
    Untracked:  output/voomQual.csv
    Untracked:  output/voomQual_treat.csv

Unstaged changes:
    Modified:   analysis/FOX_PE_RUVSeq.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/FOX_limma.Rmd</code>) and HTML (<code>docs/FOX_limma.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sabeard4/Fetal-oxygenation-study/blob/b4f08497019cf8c9634776160bfd699462eafa69/analysis/FOX_limma.Rmd" target="_blank">b4f0849</a>
</td>
<td>
Sally Beard
</td>
<td>
2022-09-27
</td>
<td>
update index and add new analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>The sequence data used in the following analysis was generated at AGRF for the <a href="https://link.springer.com/article/10.1186/s12916-020-01605-x">FOX study</a>. RNA was isolated from whole blood PAXgene tubes and Illumina’s TruSeq stranded RNA sample preparation kit with globin depletion was used to prepare libraries for sequencing. Libraries were pooled and clustered using the Illumina cBot system, followed by sequencing on the Illumina HiSeq 2000 platform to obtain 50 base pair single-end reads. Each library was split over two lanes, which generated two sequence files per sample. Fastq files were aligned against the human genome GRCh38 using <code>hisat2</code>, then files were sorted with <code>SAMtools</code>, before reads aligning to exons were counted using <code>htseq</code> with default parameters. Technical duplicates run in different sequencing lanes overlapped on MDS plots, indicating there was no batch effect due to lane, therefore in this analysis counts from technical duplicates were summed for further analysis.</p>
<pre class="r"><code>#import required libraries
library(limma)
library(Glimma)
library(edgeR)
library(tidyverse)
library(EDASeq)
library(RColorBrewer)
library(tableone)
library(knitr)
library(kableExtra)
library(Homo.sapiens)
library(patchwork)
library(NMF)
library(here)
library(ggrepel)
library(clusterProfiler)</code></pre>
<div id="data-import" class="section level2">
<h2>Data import</h2>
<p>Import sample information and create table of characteristics.</p>
<pre class="r"><code>metadata &lt;- read.csv(&quot;data/FOX_REDCap_metadata.csv&quot;)
#create a variable list
vars &lt;- c(&quot;Age&quot;, &quot;BMI&quot;, &quot;Diabetes&quot;, &quot;Hypertension&quot;,&quot;GA_sampling&quot;, &quot;GA_delivery&quot;, &quot;Birthweight&quot;, &quot;Birthweight_centile&quot;, &quot;Sex&quot;, &quot;Cord_blood_pH&quot;)

#create Table 1 stratified by case-control
tableOne &lt;- CreateTableOne(vars = vars, strata = &quot;Case&quot;, data = metadata, testNonNormal = kruskal.test)
kable_styling(kableone(tableOne, caption=&quot;Table 1: Patient characteristics for FOX cases included in RNA-seq analysis&quot;, nonnormal = c(&quot;Birthweight_centile&quot;, &quot;Cord_blood_pH&quot;, &quot;BMI&quot;), varLabels = TRUE, dropEqual = TRUE))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Table 1: Patient characteristics for FOX cases included in RNA-seq analysis
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Case
</th>
<th style="text-align:left;">
Control
</th>
<th style="text-align:left;">
p
</th>
<th style="text-align:left;">
test
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
n
</td>
<td style="text-align:left;">
129
</td>
<td style="text-align:left;">
79
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Age (mean (SD))
</td>
<td style="text-align:left;">
32.23 (5.51)
</td>
<td style="text-align:left;">
29.91 (5.76)
</td>
<td style="text-align:left;">
0.004
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
BMI (median [IQR])
</td>
<td style="text-align:left;">
25.00 [22.00, 30.00]
</td>
<td style="text-align:left;">
22.00 [21.00, 25.00]
</td>
<td style="text-align:left;">
&lt;0.001
</td>
<td style="text-align:left;">
nonnorm
</td>
</tr>
<tr>
<td style="text-align:left;">
Diabetes (%)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
0.126
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
GDM
</td>
<td style="text-align:left;">
12 ( 9.3)
</td>
<td style="text-align:left;">
4 ( 5.2)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
110 (85.3)
</td>
<td style="text-align:left;">
73 ( 94.8)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
T1DM
</td>
<td style="text-align:left;">
2 ( 1.6)
</td>
<td style="text-align:left;">
0 ( 0.0)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
T2DM
</td>
<td style="text-align:left;">
5 ( 3.9)
</td>
<td style="text-align:left;">
0 ( 0.0)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Hypertension (%)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
&lt;0.001
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Chronic_Essential_HT
</td>
<td style="text-align:left;">
12 ( 9.3)
</td>
<td style="text-align:left;">
0 ( 0.0)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Gestational_HT
</td>
<td style="text-align:left;">
10 ( 7.8)
</td>
<td style="text-align:left;">
0 ( 0.0)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
45 (34.9)
</td>
<td style="text-align:left;">
77 (100.0)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Preeclampsia
</td>
<td style="text-align:left;">
62 (48.1)
</td>
<td style="text-align:left;">
0 ( 0.0)
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
GA_sampling (mean (SD))
</td>
<td style="text-align:left;">
30.39 (2.17)
</td>
<td style="text-align:left;">
29.98 (2.21)
</td>
<td style="text-align:left;">
0.209
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
GA_delivery (mean (SD))
</td>
<td style="text-align:left;">
30.39 (2.17)
</td>
<td style="text-align:left;">
39.52 (1.03)
</td>
<td style="text-align:left;">
&lt;0.001
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Birthweight (mean (SD))
</td>
<td style="text-align:left;">
1070.09 (330.97)
</td>
<td style="text-align:left;">
3545.22 (363.38)
</td>
<td style="text-align:left;">
&lt;0.001
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Birthweight_centile (median [IQR])
</td>
<td style="text-align:left;">
5.50 [1.30, 14.70]
</td>
<td style="text-align:left;">
72.00 [61.00, 84.00]
</td>
<td style="text-align:left;">
&lt;0.001
</td>
<td style="text-align:left;">
nonnorm
</td>
</tr>
<tr>
<td style="text-align:left;">
Sex (%)
</td>
<td style="text-align:left;">
76 (58.9)
</td>
<td style="text-align:left;">
40 ( 51.9)
</td>
<td style="text-align:left;">
0.406
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Cord_blood_pH (median [IQR])
</td>
<td style="text-align:left;">
7.28 [7.23, 7.30]
</td>
<td style="text-align:left;">
7.25 [7.20, 7.28]
</td>
<td style="text-align:left;">
0.104
</td>
<td style="text-align:left;">
nonnorm
</td>
</tr>
</tbody>
</table>
<p>Set up <code>DGElist</code> object for downstream analysis and add sample-specific information.</p>
<pre class="r"><code>#list count files from working directory
files = list.files(&quot;/Users/beards/rnaseq_FOX/FOX_case_control_counts&quot;, pattern = &quot;[counts]$&quot;)

#edgeR has a function readDGE for combining separate count files into one matrix of counts in one step
counts &lt;- readDGE(files, path = &quot;/Users/beards/rnaseq_FOX/FOX_case_control_counts&quot;, columns = c(1,3), header = FALSE)

#remove meta tags as library size includes counts from the meta tags
MetaTags &lt;- grep(&quot;^__&quot;, rownames(counts))
counts &lt;- counts[-MetaTags, ]

#update library size to reflect mapped reads
counts$samples$lib.size &lt;- colSums(counts$counts)

#obtain sample names from file names
sample &lt;- strsplit2(files, &quot;_&quot;)[,1]
counts$samples$sample &lt;- as.factor(sample)
samplenames &lt;- counts$samples$sample

#sum counts from technical replicates
counts &lt;- sumTechReps(counts, ID=samplenames)

#Add sample-level information related to experimental design (and possible confounding factors) to the count matrix
counts$samples$site &lt;- as.factor(metadata$Site)
counts$samples$sample &lt;- as.factor(metadata$sample)
counts$samples$sex &lt;- as.factor(metadata$Sex)
counts$samples$hypertension &lt;- as.factor(metadata$Hypertension)
counts$samples$diabetes &lt;- as.factor(metadata$Diabetes)
counts$samples$steroids &lt;- as.factor(metadata$Antenatal_steroids)
counts$samples$group &lt;- as.factor(metadata$Case)

#import information for each gene
geneid &lt;- rownames(counts)
genes &lt;- select(Homo.sapiens, keys=geneid, columns=c(&quot;SYMBOL&quot;, &quot;TXCHROM&quot;, &quot;ENTREZID&quot;), 
                keytype=&quot;ENSEMBL&quot;)
genes &lt;- genes[!duplicated(genes$ENSEMBL),]
counts$genes &lt;- genes

#remove genes without ENTREZ IDs
keep &lt;- !is.na(counts$genes$ENTREZID) &amp; !is.null(counts$genes$ENTREZID)
counts &lt;- counts[keep, ]

#generate library statistics
cpm &lt;- cpm(counts)
lcpm &lt;- cpm(counts, log=TRUE)
L &lt;- mean(counts$samples$lib.size) * 1e-6
M &lt;- median(counts$samples$lib.size) * 1e-6</code></pre>
</div>
<div id="quality-control" class="section level2">
<h2>Quality Control</h2>
<p>Genes with no or low counts are unlikely to be biologically important, they provide little evidence for differential expression, and interfere with statistical approximations, so they were filtered out before performing the analysis. Genes with low or no counts (counts per million reads that corresponded to less than ~10 reads in any 79 samples (smallest group size)) were filtered out of the analysis, based on the recommendations in the edgeR documentation. Both smaller (&gt;2) and larger (&lt;100) thresholds were tested, lower thresholds detected more significant genes on average, higher thresholds detected less differentially expressed genes, but those that were detected are likely to be more robust. The threshold chosen for the analysis is commonly used in the literature and provided a good compromise.</p>
<pre class="r"><code>#filter out genes with low expression
keep.exprs &lt;- filterByExpr(counts, group=counts$samples$group)
counts_FOX &lt;- counts[keep.exprs,, keep.lib.sizes=FALSE]
samplenames &lt;- colnames(counts_FOX)</code></pre>
<p>Plotting the distribution of log-CPM values (below) showed that before filtering (A) many genes in each sample had low or no expression, with negative log-CPM values. These genes were removed from the analysis after filtering (B). Filtering didn’t completely improve the uniformity between samples however.</p>
<pre class="r"><code>#make density plots for counts before and after filtering out genes with low expression
par(mfrow = c(1,1))
lcpmz &lt;- lcpm
lcpm.cutoff &lt;- log2(10/M + 2/L)
nsamples &lt;- ncol(counts)
col &lt;- scales::hue_pal()(nsamples)
plot(density(lcpm[,1]), col=col[1], lwd=2, ylim=c(0,0.4), las=2, main=&quot;&quot;, xlab=&quot;&quot;)
  title(main=&quot;A. Raw data&quot;, xlab=&quot;Log-cpm&quot;)
abline(v=lcpm.cutoff, lty=3)
for (i in 2:nsamples){
  den &lt;- density(lcpm[,i])
  lines(den$x, den$y, col=col[i], lwd=2)
}</code></pre>
<p><img src="figure/FOX_limma.Rmd/denisty-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>lcpm2 &lt;- cpm(counts_FOX, log=TRUE)
plot(density(lcpm2[,1]), col=col[1], lwd=2, ylim=c(0,0.4), las=2, main=&quot;&quot;, xlab=&quot;&quot;)
title(main=&quot;B. Filtered data&quot;, xlab=&quot;Log-cpm&quot;)
abline(v=lcpm.cutoff, lty=3)
for (i in 2:nsamples){
  den &lt;- density(lcpm2[,i])
  lines(den$x, den$y, col=col[i], lwd=2)
}</code></pre>
<p><img src="figure/FOX_limma.Rmd/denisty-plot-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>We suspected that effective library size might have been contributing to the lack of uniformity between samples in the density plots. Colouring the samples by effective library size showed that the distribution of counts was similar for samples with larger libraries (8-12 million reads) where two peaks were seen after filtering, one at logCPM ~-1.5 and one at 5 (B). Samples with smaller libraries tended to have a single peak between ~3 and 5 log-CPM (B).</p>
<pre class="r"><code>#create binds for effective samples size
lib.size &lt;- counts_FOX$samples$lib.size
#set up cut-off values
breaks &lt;- c(0,2000000,4000000,6000000,8000000,10000000,12000000)
#specify bin labels
tags &lt;- c(&quot;[0-2m)&quot;, &quot;[2-4m)&quot;, &quot;[4-6m)&quot;, &quot;[6-8m)&quot;, &quot;[8-10m)&quot;, &quot;[10-12m)&quot;)
#bucket library sizes into bins
group_tags &lt;- cut(lib.size,
                  breaks=breaks,
                  include.lowest=TRUE,
                  right=FALSE,
                  labels=tags)
#add new tags to count matrix
counts_FOX$samples$lib_size &lt;- as.factor(group_tags)

#make density plots for counts before and after filtering out genes with low expression
par(mfrow = c(1,1))
lcpmz &lt;- lcpm
lcpm.cutoff &lt;- log2(10/M + 2/L)
nsamples &lt;- ncol(counts)
col.lib &lt;- counts_FOX$samples$lib_size
levels(col.lib) &lt;- brewer.pal(nlevels(col.lib), &quot;Set1&quot;)
plot(density(lcpmz[,1]), col = as.character(col.lib[1]), lwd = 1, ylim = c(0, 0.4), las = 2, 
     main = &quot;&quot;, xlab = &quot;&quot;)
title(main = &quot;A. Raw data&quot;, xlab = &quot;Log-cpm&quot;)
abline(v = lcpm.cutoff, lty = 3)
for (i in 2:nsamples){
  den &lt;- density(lcpmz[,i])
  lines(den$x, den$y, col = as.character(col.lib[i]), lwd = 1)
}
legend(&quot;topright&quot;,legend=levels(counts_FOX$samples$lib_size), col=levels(col.lib), pch=16, title =&quot;Library size (millions)&quot;)</code></pre>
<p><img src="figure/FOX_limma.Rmd/density-plot-coloured-lib.size-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>lcpmy &lt;- cpm(counts_FOX, log=TRUE)
plot(density(lcpmy[,1]), col = as.character(col.lib[1]), lwd = 1, ylim = c(0, 0.4), las = 2, 
     main = &quot;&quot;, xlab = &quot;&quot;)
title(main = &quot;B. Filtered data&quot;, xlab = &quot;Log-cpm&quot;)
abline(v = lcpm.cutoff, lty = 3)
for (i in 2:nsamples){
  den &lt;- density(lcpmy[,i])
  lines(den$x, den$y, col = as.character(col.lib[i]), lwd = 1)
}
legend(&quot;topright&quot;,legend=levels(counts_FOX$samples$lib_size), col=levels(col.lib), pch=16, title =&quot;Library size (millions)&quot; )</code></pre>
<p><img src="figure/FOX_limma.Rmd/density-plot-coloured-lib.size-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>TMM normalisation was applied to account for any highly expressed genes monopolising the sequencing and to account for the underlying differences in the distributions of expressed transcripts between samples.</p>
<pre class="r"><code>#apply TMM normalisation
norm_FOX &lt;- calcNormFactors(counts_FOX, method = &quot;TMM&quot;)</code></pre>
<p>Although ~30 million reads per sample were obtained from the sequencing, effective library sizes varied from around 2 million reads to more than 10 million reads, with the median just under 7 million reads (A). This suggests we are only likely to capture the most abundant maternal RNAs. The range and distribution of expression values look fairly uniform between samples, which is what we expect, and TMM normalisation further improves this.</p>
<pre class="r"><code>dat &lt;- data.frame(lib = norm_FOX$samples$lib.size,
                  status = norm_FOX$samples$group,
                  sample = colnames(norm_FOX))
p1 &lt;- ggplot(dat, aes(x = sample, y = lib, fill = status)) +
  geom_bar(stat = &quot;identity&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = &quot;Sample&quot;, y = &quot;Library size&quot;,
       fill = &quot;Preeclampsia status&quot;, title = &quot;A. Library size after filtering&quot;) +
  geom_hline(yintercept = median(dat$lib), linetype = &quot;dashed&quot;) +
  scale_x_discrete(limits = dat$sample)

dat &lt;- reshape2::melt(cpm(norm_FOX, normalized.lib.sizes = FALSE, log = TRUE),
                      value.name = &quot;cpm&quot;)
dat$status &lt;- rep(norm_FOX$samples$group, each = nrow(norm_FOX))
colnames(dat)[2] &lt;- &quot;sample&quot;
p2 &lt;- ggplot(dat, aes(x = sample, y = cpm, fill = status)) +
  geom_boxplot(show.legend = FALSE, outlier.size = 0.75) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7)) +
  labs(x = &quot;Sample&quot;, y = &quot;Log-CPM&quot;,
       fill = &quot;Preeclampsia status&quot;, title = &quot;B. Expression values after filtering&quot;) +
  geom_hline(yintercept = median(dat$lib), linetype = &quot;dashed&quot;)

dat &lt;- reshape2::melt(cpm(norm_FOX, normalized.lib.sizes = TRUE, log = TRUE),
                      value.name = &quot;cpm&quot;)

dat$status &lt;- rep(norm_FOX$samples$group, each = nrow(norm_FOX))
colnames(dat)[2] &lt;- &quot;sample&quot;
p3 &lt;- ggplot(dat, aes(x = sample, y = cpm, fill = status)) +
  geom_boxplot(show.legend = FALSE, outlier.size = 0.75) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7)) +
  labs(x = &quot;Sample&quot;, y = &quot;Log-CPM&quot;,
       fill = &quot;Status&quot;, title = &quot;C. Expression values after normalisation&quot;) +
  geom_hline(yintercept = median(dat$lib), linetype = &quot;dashed&quot;)

p1</code></pre>
<p><img src="figure/FOX_limma.Rmd/expression-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p2</code></pre>
<p><img src="figure/FOX_limma.Rmd/expression-plot-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p3</code></pre>
<p><img src="figure/FOX_limma.Rmd/expression-plot-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Multi-dimensional scaling (MDS) plots show the largest sources of variation in the data. They are a good way of identifying structure in the data and exploring relationships between samples. The following plots examine the first four principle components coloured for known features of the data including preeclampsia status, fetal sex, collection site, diabates status, steroid treatment prior to blood collection and library size. The samples did not cluster by known features of the data, except library size. The lack of clustering by case-control status in any of the first four principle components suggests we might not find any differentially expressed genes. The MDS plots suggest that the largest source of variation in the data is library size, as there is a strong trend towards increasingly smaller library sizes from left to right across the first dimension.</p>
<pre class="r"><code>dims &lt;- list(c(1,2), c(1,3), c(2,3), c(3,4))
vars &lt;- c(&quot;group&quot;, &quot;sex&quot;, &quot;site&quot;, &quot;diabetes&quot;, &quot;steroids&quot;, &quot;lib_size&quot;)
patches &lt;- vector(&quot;list&quot;, length(vars))

for(i in 1:length(vars)){
  p &lt;- vector(&quot;list&quot;, length(dims))
  
  for(j in 1:length(dims)){
    mds &lt;- plotMDS(cpm(norm_FOX, log = TRUE), top = 1000, gene.selection=&quot;common&quot;, 
                   plot = FALSE, dim.plot = dims[[j]])
    dat &lt;- tibble::tibble(x = mds$x, y = mds$y,
                          sample = samplenames,
                          variable = pull(norm_FOX$samples, vars[i]))
    
    p[[j]] &lt;- ggplot(dat, aes(x = x, y = y, colour = variable)) +
      geom_text(aes(label = sample), size = 2.5) +
      labs(x = glue::glue(&quot;Principal component {dims[[j]][1]}&quot;), 
           y = glue::glue(&quot;Principal component {dims[[j]][2]}&quot;),
           colour = vars[i])
  }
  
  patches[[i]] &lt;- wrap_elements(wrap_plots(p, ncol = 2, guides = &quot;collect&quot;) +
    plot_annotation(title = glue::glue(&quot;Coloured by: {vars[i]}&quot;)) &amp;
    theme(legend.position = &quot;bottom&quot;))
  
}

patches</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/FOX_limma.Rmd/mds-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/FOX_limma.Rmd/mds-plot-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/FOX_limma.Rmd/mds-plot-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/FOX_limma.Rmd/mds-plot-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/FOX_limma.Rmd/mds-plot-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[6]]</code></pre>
<p><img src="figure/FOX_limma.Rmd/mds-plot-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-expression-analysis" class="section level2">
<h2>Differential expression analysis</h2>
<p>Due to the variability in the data and the strong influence of library size, the TMM normalised data was transformed using <code>voomWithQualityWeights</code>. This takes into account the different library sizes and the mean variance relationship in the data and calculates sample-specific quality weights. Linear models were fit in <code>limma</code>, incorporating the <code>voom</code> weights. The samples from preterm growth restricted pregnancies were compared to controls. A summary of the number of differentially expressed genes (table 2) and their details (table 3) are shown below.</p>
<pre class="r"><code>#relevel the groups so that control is the reference
group &lt;- as.factor(norm_FOX$samples$group)
group &lt;- relevel(group, &quot;Control&quot;)

#set up design matrix to test case status
design1 &lt;- model.matrix(~group)
colnames(design1) &lt;- gsub(&quot;group&quot;, &quot;&quot;, colnames(design1))
rownames(design1) &lt;- samplenames

vwts &lt;- voomWithQualityWeights(norm_FOX, design1, plot=TRUE)</code></pre>
<p><img src="figure/FOX_limma.Rmd/voom-output-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>vfit &lt;- lmFit(vwts, design1)
vfit &lt;- contrasts.fit(vfit, coef = 2)
efit &lt;- eBayes(vfit)

kable_styling(kable(summary(decideTests(efit)), caption=&quot;Table 2: Number of differentially expressed genes&quot;))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Table 2: Number of differentially expressed genes
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Case
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
2692
</td>
</tr>
<tr>
<td style="text-align:left;">
NotSig
</td>
<td style="text-align:right;">
8448
</td>
</tr>
<tr>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
1967
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>top_table &lt;- topTable(efit, n=40, sort.by = &quot;P&quot;)
write.csv(topTable(efit, n=Inf), &quot;./output/voomQual.csv&quot;)
kable_styling(kable(top_table[,c(3,5,6,9)], caption=&quot;Table 3: Top 40 DE genes in cases compared to controls&quot;, format.args = list(scientific = -1), digits = 50))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Table 3: Top 40 DE genes in cases compared to controls
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
SYMBOL
</th>
<th style="text-align:right;">
logFC
</th>
<th style="text-align:right;">
AveExpr
</th>
<th style="text-align:right;">
adj.P.Val
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ENSG00000157514
</td>
<td style="text-align:left;">
TSC22D3
</td>
<td style="text-align:right;">
0.7982322
</td>
<td style="text-align:right;">
8.8528112
</td>
<td style="text-align:right;">
1.794210e-22
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000136603
</td>
<td style="text-align:left;">
SKIL
</td>
<td style="text-align:right;">
0.7073969
</td>
<td style="text-align:right;">
7.0413201
</td>
<td style="text-align:right;">
1.692867e-19
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000206560
</td>
<td style="text-align:left;">
ANKRD28
</td>
<td style="text-align:right;">
0.7370766
</td>
<td style="text-align:right;">
6.6179609
</td>
<td style="text-align:right;">
5.247883e-19
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000188596
</td>
<td style="text-align:left;">
CFAP54
</td>
<td style="text-align:right;">
1.3370782
</td>
<td style="text-align:right;">
2.4998678
</td>
<td style="text-align:right;">
1.015557e-18
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000169902
</td>
<td style="text-align:left;">
TPST1
</td>
<td style="text-align:right;">
1.2349073
</td>
<td style="text-align:right;">
4.5823962
</td>
<td style="text-align:right;">
1.698715e-18
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000227507
</td>
<td style="text-align:left;">
LTB
</td>
<td style="text-align:right;">
-0.6091819
</td>
<td style="text-align:right;">
7.0385102
</td>
<td style="text-align:right;">
3.827614e-18
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000164691
</td>
<td style="text-align:left;">
TAGAP
</td>
<td style="text-align:right;">
0.3768047
</td>
<td style="text-align:right;">
8.3831960
</td>
<td style="text-align:right;">
4.989859e-17
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000110090
</td>
<td style="text-align:left;">
CPT1A
</td>
<td style="text-align:right;">
0.5230188
</td>
<td style="text-align:right;">
5.7326497
</td>
<td style="text-align:right;">
1.161077e-15
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000118503
</td>
<td style="text-align:left;">
TNFAIP3
</td>
<td style="text-align:right;">
0.4818610
</td>
<td style="text-align:right;">
5.8884026
</td>
<td style="text-align:right;">
1.640417e-15
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000160593
</td>
<td style="text-align:left;">
JAML
</td>
<td style="text-align:right;">
-0.4855119
</td>
<td style="text-align:right;">
10.2024195
</td>
<td style="text-align:right;">
1.644373e-15
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000162191
</td>
<td style="text-align:left;">
UBXN1
</td>
<td style="text-align:right;">
-0.3553075
</td>
<td style="text-align:right;">
5.5466053
</td>
<td style="text-align:right;">
1.742011e-15
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000149573
</td>
<td style="text-align:left;">
MPZL2
</td>
<td style="text-align:right;">
-0.6853475
</td>
<td style="text-align:right;">
3.5661237
</td>
<td style="text-align:right;">
2.197832e-15
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000099337
</td>
<td style="text-align:left;">
KCNK6
</td>
<td style="text-align:right;">
-0.3425057
</td>
<td style="text-align:right;">
4.3287326
</td>
<td style="text-align:right;">
4.639178e-15
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000137818
</td>
<td style="text-align:left;">
RPLP1
</td>
<td style="text-align:right;">
-0.8731848
</td>
<td style="text-align:right;">
6.7434242
</td>
<td style="text-align:right;">
7.755002e-15
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000138670
</td>
<td style="text-align:left;">
RASGEF1B
</td>
<td style="text-align:right;">
0.4483372
</td>
<td style="text-align:right;">
4.5251865
</td>
<td style="text-align:right;">
8.808110e-15
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000175591
</td>
<td style="text-align:left;">
P2RY2
</td>
<td style="text-align:right;">
-0.7826471
</td>
<td style="text-align:right;">
1.8765960
</td>
<td style="text-align:right;">
1.074433e-14
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000103056
</td>
<td style="text-align:left;">
SMPD3
</td>
<td style="text-align:right;">
-1.0762800
</td>
<td style="text-align:right;">
2.0513535
</td>
<td style="text-align:right;">
1.750405e-14
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000101160
</td>
<td style="text-align:left;">
CTSZ
</td>
<td style="text-align:right;">
-0.4446875
</td>
<td style="text-align:right;">
7.1798659
</td>
<td style="text-align:right;">
1.947531e-14
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000135486
</td>
<td style="text-align:left;">
HNRNPA1
</td>
<td style="text-align:right;">
-0.5585863
</td>
<td style="text-align:right;">
6.7990365
</td>
<td style="text-align:right;">
1.976056e-14
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000004799
</td>
<td style="text-align:left;">
PDK4
</td>
<td style="text-align:right;">
0.7758523
</td>
<td style="text-align:right;">
5.0210031
</td>
<td style="text-align:right;">
3.249060e-14
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000108819
</td>
<td style="text-align:left;">
PPP1R9B
</td>
<td style="text-align:right;">
-0.4491144
</td>
<td style="text-align:right;">
6.2752888
</td>
<td style="text-align:right;">
4.727486e-14
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000213085
</td>
<td style="text-align:left;">
CFAP45
</td>
<td style="text-align:right;">
0.9481530
</td>
<td style="text-align:right;">
2.9366665
</td>
<td style="text-align:right;">
8.713705e-14
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000183134
</td>
<td style="text-align:left;">
PTGDR2
</td>
<td style="text-align:right;">
-1.1514718
</td>
<td style="text-align:right;">
2.0782962
</td>
<td style="text-align:right;">
9.772841e-14
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000160087
</td>
<td style="text-align:left;">
UBE2J2
</td>
<td style="text-align:right;">
-0.2353195
</td>
<td style="text-align:right;">
5.1186270
</td>
<td style="text-align:right;">
1.545453e-13
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000151694
</td>
<td style="text-align:left;">
ADAM17
</td>
<td style="text-align:right;">
0.3984995
</td>
<td style="text-align:right;">
6.5675089
</td>
<td style="text-align:right;">
2.023379e-13
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000278099
</td>
<td style="text-align:left;">
RNVU1-2A
</td>
<td style="text-align:right;">
-1.1795871
</td>
<td style="text-align:right;">
6.5563456
</td>
<td style="text-align:right;">
2.980208e-13
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000135390
</td>
<td style="text-align:left;">
ATP5MC2
</td>
<td style="text-align:right;">
-0.3670185
</td>
<td style="text-align:right;">
5.5647590
</td>
<td style="text-align:right;">
4.021539e-13
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000096093
</td>
<td style="text-align:left;">
EFHC1
</td>
<td style="text-align:right;">
0.5628333
</td>
<td style="text-align:right;">
3.9137591
</td>
<td style="text-align:right;">
5.834055e-13
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000134248
</td>
<td style="text-align:left;">
LAMTOR5
</td>
<td style="text-align:right;">
-0.4211873
</td>
<td style="text-align:right;">
4.7149935
</td>
<td style="text-align:right;">
5.853354e-13
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000239961
</td>
<td style="text-align:left;">
LILRA4
</td>
<td style="text-align:right;">
-0.8411866
</td>
<td style="text-align:right;">
0.4056092
</td>
<td style="text-align:right;">
5.997310e-13
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000119655
</td>
<td style="text-align:left;">
NPC2
</td>
<td style="text-align:right;">
-0.4366324
</td>
<td style="text-align:right;">
5.9205634
</td>
<td style="text-align:right;">
7.418052e-13
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000162881
</td>
<td style="text-align:left;">
OXER1
</td>
<td style="text-align:right;">
-0.6839051
</td>
<td style="text-align:right;">
2.1154811
</td>
<td style="text-align:right;">
7.731401e-13
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000026508
</td>
<td style="text-align:left;">
CD44
</td>
<td style="text-align:right;">
0.3748140
</td>
<td style="text-align:right;">
8.9093234
</td>
<td style="text-align:right;">
9.012436e-13
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000178982
</td>
<td style="text-align:left;">
EIF3K
</td>
<td style="text-align:right;">
-0.4588080
</td>
<td style="text-align:right;">
6.0914669
</td>
<td style="text-align:right;">
1.183409e-12
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000163659
</td>
<td style="text-align:left;">
TIPARP
</td>
<td style="text-align:right;">
0.3310633
</td>
<td style="text-align:right;">
6.3437673
</td>
<td style="text-align:right;">
1.264883e-12
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000160888
</td>
<td style="text-align:left;">
IER2
</td>
<td style="text-align:right;">
-0.4159545
</td>
<td style="text-align:right;">
5.7152090
</td>
<td style="text-align:right;">
1.426207e-12
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000170906
</td>
<td style="text-align:left;">
NDUFA3
</td>
<td style="text-align:right;">
-0.3860299
</td>
<td style="text-align:right;">
3.9346154
</td>
<td style="text-align:right;">
1.720716e-12
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000171161
</td>
<td style="text-align:left;">
ZNF672
</td>
<td style="text-align:right;">
-0.2737535
</td>
<td style="text-align:right;">
4.3874747
</td>
<td style="text-align:right;">
2.205336e-12
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000196511
</td>
<td style="text-align:left;">
TPK1
</td>
<td style="text-align:right;">
0.4212962
</td>
<td style="text-align:right;">
5.6063845
</td>
<td style="text-align:right;">
2.439580e-12
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000134531
</td>
<td style="text-align:left;">
EMP1
</td>
<td style="text-align:right;">
1.3480508
</td>
<td style="text-align:right;">
1.0828680
</td>
<td style="text-align:right;">
3.013891e-12
</td>
</tr>
</tbody>
</table>
<p>Since so many DE genes were identified, a logFC cut-off was applied using the <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2654802/"><code>treat</code></a> method, which can be used to calculate p-values from empirical Bayes moderated t-statistics with a minimum log-FC requirement. The cutoff was set to 0.5, which is means that only genes that have at least a 1.4-fold change in expression in cases compared to controls will be detected. A summary of the number of differentially expressed genes (table 3) and their details (table 4) are shown below.</p>
<pre class="r"><code>tfit &lt;- treat(vfit, lfc=0.5)
kable_styling(kable(summary(decideTests(tfit)), caption=&quot;Table 3: Number of differentially expressed genes from treat method&quot;))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Table 3: Number of differentially expressed genes from treat method
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Case
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
NotSig
</td>
<td style="text-align:right;">
13087
</td>
</tr>
<tr>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
12
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>top_treat &lt;- data.frame(topTreat(tfit, n=20, sort.by = &quot;P&quot;))
write.csv(top_treat, &quot;./output/voomQual_treat.csv&quot;)
kable_styling(kable(top_treat[,c(3,5,6,9)], caption=&quot;Table 4: DE genes in cases compared to controls with treat method&quot;, format.args = list(scientific = -1), digits = 50))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Table 4: DE genes in cases compared to controls with treat method
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
SYMBOL
</th>
<th style="text-align:right;">
logFC
</th>
<th style="text-align:right;">
AveExpr
</th>
<th style="text-align:right;">
adj.P.Val
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ENSG00000188596
</td>
<td style="text-align:left;">
CFAP54
</td>
<td style="text-align:right;">
1.3370782
</td>
<td style="text-align:right;">
2.4998678
</td>
<td style="text-align:right;">
6.074390e-07
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000169902
</td>
<td style="text-align:left;">
TPST1
</td>
<td style="text-align:right;">
1.2349073
</td>
<td style="text-align:right;">
4.5823962
</td>
<td style="text-align:right;">
2.850583e-06
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000087116
</td>
<td style="text-align:left;">
ADAMTS2
</td>
<td style="text-align:right;">
2.9768696
</td>
<td style="text-align:right;">
0.3365088
</td>
<td style="text-align:right;">
1.695745e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000146122
</td>
<td style="text-align:left;">
DAAM2
</td>
<td style="text-align:right;">
2.0878744
</td>
<td style="text-align:right;">
2.6766188
</td>
<td style="text-align:right;">
1.695745e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000115414
</td>
<td style="text-align:left;">
FN1
</td>
<td style="text-align:right;">
2.6409061
</td>
<td style="text-align:right;">
0.2545851
</td>
<td style="text-align:right;">
4.106956e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000155659
</td>
<td style="text-align:left;">
VSIG4
</td>
<td style="text-align:right;">
1.7167547
</td>
<td style="text-align:right;">
2.0329477
</td>
<td style="text-align:right;">
8.643437e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000205927
</td>
<td style="text-align:left;">
OLIG2
</td>
<td style="text-align:right;">
-1.4832119
</td>
<td style="text-align:right;">
-0.1465995
</td>
<td style="text-align:right;">
1.817892e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000134531
</td>
<td style="text-align:left;">
EMP1
</td>
<td style="text-align:right;">
1.3480508
</td>
<td style="text-align:right;">
1.0828680
</td>
<td style="text-align:right;">
3.000850e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000250696
</td>
<td style="text-align:left;">
LOC105377267
</td>
<td style="text-align:right;">
-1.4923254
</td>
<td style="text-align:right;">
1.2439405
</td>
<td style="text-align:right;">
3.000850e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000154330
</td>
<td style="text-align:left;">
PGM5
</td>
<td style="text-align:right;">
1.6118278
</td>
<td style="text-align:right;">
1.6611036
</td>
<td style="text-align:right;">
3.317035e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000183134
</td>
<td style="text-align:left;">
PTGDR2
</td>
<td style="text-align:right;">
-1.1514718
</td>
<td style="text-align:right;">
2.0782962
</td>
<td style="text-align:right;">
5.120460e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000278099
</td>
<td style="text-align:left;">
RNVU1-2A
</td>
<td style="text-align:right;">
-1.1795871
</td>
<td style="text-align:right;">
6.5563456
</td>
<td style="text-align:right;">
5.120460e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000103056
</td>
<td style="text-align:left;">
SMPD3
</td>
<td style="text-align:right;">
-1.0762800
</td>
<td style="text-align:right;">
2.0513535
</td>
<td style="text-align:right;">
6.873087e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000161905
</td>
<td style="text-align:left;">
ALOX15
</td>
<td style="text-align:right;">
-1.4661430
</td>
<td style="text-align:right;">
1.2105093
</td>
<td style="text-align:right;">
6.873087e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000157514
</td>
<td style="text-align:left;">
TSC22D3
</td>
<td style="text-align:right;">
0.7982322
</td>
<td style="text-align:right;">
8.8528112
</td>
<td style="text-align:right;">
3.228139e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000152463
</td>
<td style="text-align:left;">
OLAH
</td>
<td style="text-align:right;">
1.9090010
</td>
<td style="text-align:right;">
0.4888689
</td>
<td style="text-align:right;">
3.438304e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000125740
</td>
<td style="text-align:left;">
FOSB
</td>
<td style="text-align:right;">
1.3370149
</td>
<td style="text-align:right;">
-0.1413510
</td>
<td style="text-align:right;">
9.024972e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000213085
</td>
<td style="text-align:left;">
CFAP45
</td>
<td style="text-align:right;">
0.9481530
</td>
<td style="text-align:right;">
2.9366665
</td>
<td style="text-align:right;">
1.183157e-02
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000125780
</td>
<td style="text-align:left;">
TGM3
</td>
<td style="text-align:right;">
-0.9818502
</td>
<td style="text-align:right;">
2.6266076
</td>
<td style="text-align:right;">
2.168699e-02
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000137818
</td>
<td style="text-align:left;">
RPLP1
</td>
<td style="text-align:right;">
-0.8731848
</td>
<td style="text-align:right;">
6.7434242
</td>
<td style="text-align:right;">
2.605078e-02
</td>
</tr>
</tbody>
</table>
<p>The volcano plot below shows the statistical significance versus the logFC of genes in the differential expression analysis with the genes identified using the treat method represented in red.</p>
<pre class="r"><code>topTreat(tfit, num = Inf) %&gt;% 
  mutate(sig = ifelse(adj.P.Val &lt;= 0.05, &quot;&lt;= 0.05&quot;, &quot;&gt; 0.05&quot;)) -&gt; dat

ggplot(dat, aes(x = logFC, y = -log10(P.Value), color = sig)) +
  geom_point(alpha = 0.75) +
  ggrepel::geom_text_repel(data = subset(dat, adj.P.Val &lt; 0.05), 
            aes(x = logFC, y = -log10(P.Value), 
                label = SYMBOL), 
            size = 2, colour = &quot;black&quot;, max.overlaps = 15) +
  labs(x = expression(~log[2]~&quot;(Fold Change)&quot;), 
       y = expression(~-log[10]~&quot;(P-value)&quot;),
       colour = &quot;FDR&quot;) +
  scale_colour_brewer(palette = &quot;Set1&quot;)</code></pre>
<p><img src="figure/FOX_limma.Rmd/volcano-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The following plots show the expression of the top 12 differentially expressed genes for samples collected from cases compared to controls. Despite the variability in the data and the lack of separation in the MDS plots, we were still able to detect differentially expressed genes, with varying degrees of overlap in expression.</p>
<pre class="r"><code>dat &lt;- reshape2::melt(cpm(norm_FOX, log = TRUE),
                      value.name = &quot;cpm&quot;)
dat$group &lt;- rep(norm_FOX$samples$group, each = nrow(norm_FOX))
dat$group &lt;- relevel(dat$group, &quot;Control&quot;)
dat$gene &lt;- rep(norm_FOX$genes$SYMBOL, ncol(norm_FOX))

p &lt;- vector(&quot;list&quot;, 12)

for(i in 1:length(p)){
  p[[i]] &lt;-ggplot(data = subset(dat, dat$gene == top_treat$SYMBOL[i]), 
       aes(x = group, y = cpm, colour = group)) +
  geom_jitter(width = 0.25) +
  stat_summary(fun = &quot;mean&quot;, geom = &quot;crossbar&quot;) +
  labs(x = &quot;Status&quot;, y = &quot;log2 Norm. CPM&quot;, colour = &quot;Status&quot;) +
  ggtitle(top_treat$SYMBOL[i]) +
  theme(plot.title = element_text(size = 8),
        plot.subtitle = element_text(size = 7),
        axis.title = element_text(size = 8),
        axis.text.x = element_text(size = 7))
}

wrap_plots(p, guides = &quot;collect&quot;, ncol = 4) &amp; 
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/FOX_limma.Rmd/gene-expression-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-set-enrichment-analysis" class="section level2">
<h2>Gene set enrichment analysis</h2>
<p>Testing for enrichment of Gene Ontology (GO) categories among statistically significant differentially expressed genes (from treat analysis) using <code>goana</code> didn’t yield any significant GO categories after adjusting for multiple testing.</p>
<pre class="r"><code>go &lt;- goana(top_treat$ENTREZID[top_treat$adj.P.Val &lt; 0.05], universe = norm_FOX$genes$entrezid)
goana_treat &lt;- topGO(go, number = Inf) %&gt;%
  mutate(FDR = p.adjust(P.DE)) %&gt;%
  dplyr::filter(FDR &lt; 0.05)
write.csv(goana_treat, &quot;./output/goana_treat.csv&quot;, row.names = TRUE)
kable_styling(knitr::kable(goana_treat, format.args = list(scientific = -1), digits = 50))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Term
</th>
<th style="text-align:left;">
Ont
</th>
<th style="text-align:right;">
N
</th>
<th style="text-align:right;">
DE
</th>
<th style="text-align:right;">
P.DE
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
</tr>
</tbody>
</table>
<p>Since GO categories overlap, the approach outlined above may be too stringent, so we also used the over-representation test implemented in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3339379/"><code>clusterProfiler</code></a> among the genes identified in the treat analysis, but only two categories were identified (below).</p>
<pre class="r"><code>gene &lt;- top_treat$ENTREZID
universe &lt;- norm_FOX$genes$ENTREZID
ego &lt;- enrichGO(gene          = gene,
                universe      = universe,
                OrgDb         = org.Hs.eg.db,
                ont           = &quot;ALL&quot;,
                pAdjustMethod = &quot;BH&quot;,
                qvalueCutoff  = 0.05,
        readable      = TRUE)
dotplot(ego)</code></pre>
<p><img src="figure/FOX_limma.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>When we tested for enrichment of GO categories among all statistically significant genes (regardless of logFC) using goana, many more categories were identified (tbale 5), but they were not very specific.</p>
<pre class="r"><code>top_table &lt;- data.frame(topTable(efit, number = Inf, sort.by = &quot;P&quot;))
go &lt;- goana(top_table$ENTREZID[top_table$adj.P.Val &lt; 0.05], universe = norm_FOX$genes$entrezid)
go_all &lt;- topGO(go, number = Inf) %&gt;%
  mutate(FDR = p.adjust(P.DE)) %&gt;%
  dplyr::filter(FDR &lt; 0.05)
write.csv(go_all, &quot;./output/goana_all.csv&quot;, row.names = TRUE)
kable_styling(knitr::kable(go_all[1:40,], format.args = list(scientific = -1), digits = 50, caption=&quot;Table 5: Top 40 GO categories from goana gene set enrichment analysis using all genes with FDR &lt; 0.05&quot;))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Table 5: Top 40 GO categories from goana gene set enrichment analysis using all genes with FDR &lt; 0.05
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Term
</th>
<th style="text-align:left;">
Ont
</th>
<th style="text-align:right;">
N
</th>
<th style="text-align:right;">
DE
</th>
<th style="text-align:right;">
P.DE
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0005737" class="uri">GO:0005737</a>
</td>
<td style="text-align:left;">
cytoplasm
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
11625
</td>
<td style="text-align:right;">
3081
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0005622" class="uri">GO:0005622</a>
</td>
<td style="text-align:left;">
intracellular
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
15073
</td>
<td style="text-align:right;">
3706
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044424" class="uri">GO:0044424</a>
</td>
<td style="text-align:left;">
intracellular part
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
15063
</td>
<td style="text-align:right;">
3703
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044444" class="uri">GO:0044444</a>
</td>
<td style="text-align:left;">
cytoplasmic part
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
9735
</td>
<td style="text-align:right;">
2661
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0005515" class="uri">GO:0005515</a>
</td>
<td style="text-align:left;">
protein binding
</td>
<td style="text-align:left;">
MF
</td>
<td style="text-align:right;">
11807
</td>
<td style="text-align:right;">
3042
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044446" class="uri">GO:0044446</a>
</td>
<td style="text-align:left;">
intracellular organelle part
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
9811
</td>
<td style="text-align:right;">
2627
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044422" class="uri">GO:0044422</a>
</td>
<td style="text-align:left;">
organelle part
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
10112
</td>
<td style="text-align:right;">
2687
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043226" class="uri">GO:0043226</a>
</td>
<td style="text-align:left;">
organelle
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
14144
</td>
<td style="text-align:right;">
3456
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043229" class="uri">GO:0043229</a>
</td>
<td style="text-align:left;">
intracellular organelle
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
13294
</td>
<td style="text-align:right;">
3294
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043227" class="uri">GO:0043227</a>
</td>
<td style="text-align:left;">
membrane-bounded organelle
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
13019
</td>
<td style="text-align:right;">
3222
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043231" class="uri">GO:0043231</a>
</td>
<td style="text-align:left;">
intracellular membrane-bounded organelle
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
11488
</td>
<td style="text-align:right;">
2904
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0005488" class="uri">GO:0005488</a>
</td>
<td style="text-align:left;">
binding
</td>
<td style="text-align:left;">
MF
</td>
<td style="text-align:right;">
15267
</td>
<td style="text-align:right;">
3622
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0003674" class="uri">GO:0003674</a>
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:left;">
MF
</td>
<td style="text-align:right;">
17697
</td>
<td style="text-align:right;">
4029
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0005829" class="uri">GO:0005829</a>
</td>
<td style="text-align:left;">
cytosol
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
5085
</td>
<td style="text-align:right;">
1479
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
0.000000e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044464" class="uri">GO:0044464</a>
</td>
<td style="text-align:left;">
cell part
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
17521
</td>
<td style="text-align:right;">
3990
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
2.800000e-49
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0005623" class="uri">GO:0005623</a>
</td>
<td style="text-align:left;">
cell
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
17590
</td>
<td style="text-align:right;">
3999
</td>
<td style="text-align:right;">
0.000000e+00
</td>
<td style="text-align:right;">
2.000000e-48
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0005654" class="uri">GO:0005654</a>
</td>
<td style="text-align:left;">
nucleoplasm
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
3512
</td>
<td style="text-align:right;">
1054
</td>
<td style="text-align:right;">
2.007779e-43
</td>
<td style="text-align:right;">
4.596810e-39
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070013" class="uri">GO:0070013</a>
</td>
<td style="text-align:left;">
intracellular organelle lumen
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
5740
</td>
<td style="text-align:right;">
1558
</td>
<td style="text-align:right;">
4.225385e-39
</td>
<td style="text-align:right;">
9.673597e-35
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031974" class="uri">GO:0031974</a>
</td>
<td style="text-align:left;">
membrane-enclosed lumen
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
5740
</td>
<td style="text-align:right;">
1558
</td>
<td style="text-align:right;">
4.225385e-39
</td>
<td style="text-align:right;">
9.673597e-35
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043233" class="uri">GO:0043233</a>
</td>
<td style="text-align:left;">
organelle lumen
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
5740
</td>
<td style="text-align:right;">
1558
</td>
<td style="text-align:right;">
4.225385e-39
</td>
<td style="text-align:right;">
9.673597e-35
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071840" class="uri">GO:0071840</a>
</td>
<td style="text-align:left;">
cellular component organization or biogenesis
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
6697
</td>
<td style="text-align:right;">
1762
</td>
<td style="text-align:right;">
8.214550e-37
</td>
<td style="text-align:right;">
1.880393e-32
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006996" class="uri">GO:0006996</a>
</td>
<td style="text-align:left;">
organelle organization
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
3885
</td>
<td style="text-align:right;">
1102
</td>
<td style="text-align:right;">
1.389829e-33
</td>
<td style="text-align:right;">
3.181319e-29
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044237" class="uri">GO:0044237</a>
</td>
<td style="text-align:left;">
cellular metabolic process
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
11301
</td>
<td style="text-align:right;">
2728
</td>
<td style="text-align:right;">
2.297847e-33
</td>
<td style="text-align:right;">
5.259542e-29
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016043" class="uri">GO:0016043</a>
</td>
<td style="text-align:left;">
cellular component organization
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
6512
</td>
<td style="text-align:right;">
1702
</td>
<td style="text-align:right;">
3.363145e-33
</td>
<td style="text-align:right;">
7.697566e-29
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0012505" class="uri">GO:0012505</a>
</td>
<td style="text-align:left;">
endomembrane system
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
4441
</td>
<td style="text-align:right;">
1218
</td>
<td style="text-align:right;">
1.310103e-30
</td>
<td style="text-align:right;">
2.998432e-26
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031090" class="uri">GO:0031090</a>
</td>
<td style="text-align:left;">
organelle membrane
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
2903
</td>
<td style="text-align:right;">
852
</td>
<td style="text-align:right;">
2.505979e-30
</td>
<td style="text-align:right;">
5.735183e-26
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070727" class="uri">GO:0070727</a>
</td>
<td style="text-align:left;">
cellular macromolecule localization
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
1883
</td>
<td style="text-align:right;">
595
</td>
<td style="text-align:right;">
1.834951e-29
</td>
<td style="text-align:right;">
4.199285e-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1901564" class="uri">GO:1901564</a>
</td>
<td style="text-align:left;">
organonitrogen compound metabolic process
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
7040
</td>
<td style="text-align:right;">
1798
</td>
<td style="text-align:right;">
2.001301e-29
</td>
<td style="text-align:right;">
4.579776e-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034613" class="uri">GO:0034613</a>
</td>
<td style="text-align:left;">
cellular protein localization
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
1872
</td>
<td style="text-align:right;">
592
</td>
<td style="text-align:right;">
2.067681e-29
</td>
<td style="text-align:right;">
4.731474e-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0003824" class="uri">GO:0003824</a>
</td>
<td style="text-align:left;">
catalytic activity
</td>
<td style="text-align:left;">
MF
</td>
<td style="text-align:right;">
5710
</td>
<td style="text-align:right;">
1498
</td>
<td style="text-align:right;">
9.809033e-29
</td>
<td style="text-align:right;">
2.244503e-24
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044428" class="uri">GO:0044428</a>
</td>
<td style="text-align:left;">
nuclear part
</td>
<td style="text-align:left;">
CC
</td>
<td style="text-align:right;">
5023
</td>
<td style="text-align:right;">
1342
</td>
<td style="text-align:right;">
1.074645e-28
</td>
<td style="text-align:right;">
2.458895e-24
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051641" class="uri">GO:0051641</a>
</td>
<td style="text-align:left;">
cellular localization
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
2927
</td>
<td style="text-align:right;">
850
</td>
<td style="text-align:right;">
1.467337e-28
</td>
<td style="text-align:right;">
3.357267e-24
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044260" class="uri">GO:0044260</a>
</td>
<td style="text-align:left;">
cellular macromolecule metabolic process
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
8352
</td>
<td style="text-align:right;">
2078
</td>
<td style="text-align:right;">
1.877070e-28
</td>
<td style="text-align:right;">
4.294549e-24
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0008104" class="uri">GO:0008104</a>
</td>
<td style="text-align:left;">
protein localization
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
2765
</td>
<td style="text-align:right;">
810
</td>
<td style="text-align:right;">
2.110326e-28
</td>
<td style="text-align:right;">
4.828003e-24
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0036230" class="uri">GO:0036230</a>
</td>
<td style="text-align:left;">
granulocyte activation
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
504
</td>
<td style="text-align:right;">
212
</td>
<td style="text-align:right;">
3.949399e-27
</td>
<td style="text-align:right;">
9.035041e-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002283" class="uri">GO:0002283</a>
</td>
<td style="text-align:left;">
neutrophil activation involved in immune response
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
488
</td>
<td style="text-align:right;">
207
</td>
<td style="text-align:right;">
4.374210e-27
</td>
<td style="text-align:right;">
1.000644e-22
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044238" class="uri">GO:0044238</a>
</td>
<td style="text-align:left;">
primary metabolic process
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
11287
</td>
<td style="text-align:right;">
2686
</td>
<td style="text-align:right;">
1.276751e-26
</td>
<td style="text-align:right;">
2.920568e-22
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042119" class="uri">GO:0042119</a>
</td>
<td style="text-align:left;">
neutrophil activation
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
498
</td>
<td style="text-align:right;">
209
</td>
<td style="text-align:right;">
1.323017e-26
</td>
<td style="text-align:right;">
3.026268e-22
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043312" class="uri">GO:0043312</a>
</td>
<td style="text-align:left;">
neutrophil degranulation
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
485
</td>
<td style="text-align:right;">
205
</td>
<td style="text-align:right;">
1.355950e-26
</td>
<td style="text-align:right;">
3.101465e-22
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002446" class="uri">GO:0002446</a>
</td>
<td style="text-align:left;">
neutrophil mediated immunity
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:right;">
499
</td>
<td style="text-align:right;">
209
</td>
<td style="text-align:right;">
1.812046e-26
</td>
<td style="text-align:right;">
4.144513e-22
</td>
</tr>
</tbody>
</table>
<p>Using <code>clusterProfiler</code> with this larger set of genes as input yielded 10 significant GO categories related to neutrophils (below).</p>
<pre class="r"><code>gene &lt;- top_table$ENTREZID[top_table$adj.P.Val &lt; 0.05]
universe &lt;- norm_FOX$genes$ENTREZID
ego &lt;- enrichGO(gene          = gene,
                universe      = universe,
                OrgDb         = org.Hs.eg.db,
                ont           = &quot;ALL&quot;,
                pAdjustMethod = &quot;BH&quot;,
                qvalueCutoff  = 0.05,
        readable      = TRUE)
dotplot(ego)</code></pre>
<p><img src="figure/FOX_limma.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The <code>camera</code> function performs a competitive test to find if genes in a given set are highly ranked in terms of differential expression compared to genes that are not in that set. In the output below we tested gene sets from the Molecular Signatures Database (<a href="https://www.gsea-msigdb.org/gsea/msigdb/index.jsp_">MSigDB</a>).</p>
<p>MSigDB genesets from version 7.1 in R format were downloaded <a href="https://bioinf.wehi.edu.au/MSigDB/v7.1/">here</a> for this analysis. We looked at GO gene sets, Hallmark gene sets and curated gene sets.</p>
<p>The GO gene sets consist of genes annotated by the same GO terms.</p>
<pre class="r"><code>GO &lt;- readRDS(&quot;./data/MSigDB/Hs.c5.all.v7.1.entrez.rds&quot;, refhook = NULL)
idx &lt;- ids2indices(GO, id=norm_FOX$genes$ENTREZID, remove.empty=TRUE)
cam_GO &lt;- camera(vwts,idx,design1)
write.csv(cam_GO[cam_GO$FDR &lt; 0.05,], &quot;./output/camera_GO.csv&quot;, row.names = TRUE)
kable_styling(kable(cam_GO[cam_GO$FDR &lt; 0.05,][1:30,], format.args = list(scientific = -1), digits = 50, caption=&quot;Table 6: Camera gene set test results for MSigDB GO gene sets&quot;))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Table 6: Camera gene set test results for MSigDB GO gene sets
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
NGenes
</th>
<th style="text-align:left;">
Direction
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
GO_ALTERNATIVE_MRNA_SPLICING_VIA_SPLICEOSOME
</td>
<td style="text-align:right;">
64
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
3.602768e-08
</td>
<td style="text-align:right;">
2.203883e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_FICOLIN_1_RICH_GRANULE
</td>
<td style="text-align:right;">
120
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
4.358515e-08
</td>
<td style="text-align:right;">
2.203883e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_REGULATION_OF_MRNA_SPLICING_VIA_SPLICEOSOME
</td>
<td style="text-align:right;">
85
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
8.996509e-08
</td>
<td style="text-align:right;">
3.032723e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_RNA_SPLICING_VIA_TRANSESTERIFICATION_REACTIONS
</td>
<td style="text-align:right;">
311
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
2.062286e-07
</td>
<td style="text-align:right;">
5.213974e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_REGULATION_OF_MRNA_PROCESSING
</td>
<td style="text-align:right;">
122
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
4.284093e-07
</td>
<td style="text-align:right;">
8.665007e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_RNA_SPLICING
</td>
<td style="text-align:right;">
391
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
9.076751e-07
</td>
<td style="text-align:right;">
1.327985e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_SPLICEOSOMAL_COMPLEX
</td>
<td style="text-align:right;">
179
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
9.192028e-07
</td>
<td style="text-align:right;">
1.327985e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_REGULATION_OF_MRNA_METABOLIC_PROCESS
</td>
<td style="text-align:right;">
281
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.704189e-06
</td>
<td style="text-align:right;">
2.154308e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_MRNA_PROCESSING
</td>
<td style="text-align:right;">
451
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
3.393856e-06
</td>
<td style="text-align:right;">
3.535399e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_FORMATION_OF_CYTOPLASMIC_TRANSLATION_INITIATION_COMPLEX
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
3.831214e-06
</td>
<td style="text-align:right;">
3.535399e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_NEGATIVE_REGULATION_OF_MRNA_SPLICING_VIA_SPLICEOSOME
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
4.180426e-06
</td>
<td style="text-align:right;">
3.535399e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_EUKARYOTIC_48S_PREINITIATION_COMPLEX
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
4.195074e-06
</td>
<td style="text-align:right;">
3.535399e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_EUKARYOTIC_TRANSLATION_INITIATION_FACTOR_3_COMPLEX
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
5.714089e-06
</td>
<td style="text-align:right;">
4.392432e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_TRANSLATION_PREINITIATION_COMPLEX
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
6.450096e-06
</td>
<td style="text-align:right;">
4.392432e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_REGULATION_OF_CELLULAR_AMINO_ACID_METABOLIC_PROCESS
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
6.515028e-06
</td>
<td style="text-align:right;">
4.392432e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_REGULATION_OF_RNA_SPLICING
</td>
<td style="text-align:right;">
116
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
9.436829e-06
</td>
<td style="text-align:right;">
5.630828e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_RIBONUCLEOPROTEIN_COMPLEX
</td>
<td style="text-align:right;">
630
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
9.843442e-06
</td>
<td style="text-align:right;">
5.630828e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_CYTOPLASMIC_TRANSLATIONAL_INITIATION
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.002224e-05
</td>
<td style="text-align:right;">
5.630828e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_NEGATIVE_REGULATION_OF_MRNA_PROCESSING
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.234286e-05
</td>
<td style="text-align:right;">
6.476145e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_TRANSLATIONAL_INITIATION
</td>
<td style="text-align:right;">
179
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.280756e-05
</td>
<td style="text-align:right;">
6.476145e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_RIBOSOME
</td>
<td style="text-align:right;">
212
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.445756e-05
</td>
<td style="text-align:right;">
6.962347e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_EXTRACELLULAR_MATRIX_STRUCTURAL_CONSTITUENT
</td>
<td style="text-align:right;">
59
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
1.707274e-05
</td>
<td style="text-align:right;">
7.848026e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_PROTEASOME_ACCESSORY_COMPLEX
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
2.188853e-05
</td>
<td style="text-align:right;">
9.144104e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_TRANSLATION_INITIATION_FACTOR_ACTIVITY
</td>
<td style="text-align:right;">
46
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
2.232266e-05
</td>
<td style="text-align:right;">
9.144104e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_MRNA_METABOLIC_PROCESS
</td>
<td style="text-align:right;">
753
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
2.260482e-05
</td>
<td style="text-align:right;">
9.144104e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_CILIARY_PLASM
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
2.580426e-05
</td>
<td style="text-align:right;">
1.003686e-02
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_INTRACILIARY_TRANSPORT_PARTICLE
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
2.854866e-05
</td>
<td style="text-align:right;">
1.033235e-02
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_ANTIGEN_PROCESSING_AND_PRESENTATION_OF_EXOGENOUS_PEPTIDE_ANTIGEN_VIA_MHC_CLASS_I
</td>
<td style="text-align:right;">
72
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
2.860732e-05
</td>
<td style="text-align:right;">
1.033235e-02
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_MITOCHONDRIAL_PROTEIN_COMPLEX
</td>
<td style="text-align:right;">
243
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
3.242939e-05
</td>
<td style="text-align:right;">
1.087862e-02
</td>
</tr>
<tr>
<td style="text-align:left;">
GO_TRANSLATION_FACTOR_ACTIVITY_RNA_BINDING
</td>
<td style="text-align:right;">
75
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
3.292013e-05
</td>
<td style="text-align:right;">
1.087862e-02
</td>
</tr>
</tbody>
</table>
<p>The Hallmark gene sets are coherently expressed signatures derived by aggregating many MSigDB gene sets to represent well-defined biological states or processes.</p>
<pre class="r"><code>hallmark &lt;- readRDS(&quot;./data/MSigDB/Hs.h.all.v7.1.entrez.rds&quot;, refhook = NULL)
idx &lt;- ids2indices(hallmark, id=norm_FOX$genes$ENTREZID, remove.empty=TRUE)
cam_h &lt;- camera(vwts,idx,design1)
write.csv(cam_h[cam_h$FDR &lt; 0.05,], &quot;./output/camera_hallmark.csv&quot;, row.names = TRUE)
kable_styling(kable(cam_h[cam_h$FDR &lt; 0.05,], format.args = list(scientific = -1), digits = 50, caption=&quot;Table 7: Camera gene set test results for MSigDB Hallmark gene sets&quot;))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Table 7: Camera gene set test results for MSigDB Hallmark gene sets
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
NGenes
</th>
<th style="text-align:left;">
Direction
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
HALLMARK_MYC_TARGETS_V1
</td>
<td style="text-align:right;">
197
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.273615e-06
</td>
<td style="text-align:right;">
6.368073e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_OXIDATIVE_PHOSPHORYLATION
</td>
<td style="text-align:right;">
195
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
5.353639e-06
</td>
<td style="text-align:right;">
1.338410e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY
</td>
<td style="text-align:right;">
47
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.565794e-04
</td>
<td style="text-align:right;">
2.609657e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_UV_RESPONSE_DN
</td>
<td style="text-align:right;">
113
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
3.518120e-04
</td>
<td style="text-align:right;">
3.746947e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_KRAS_SIGNALING_DN
</td>
<td style="text-align:right;">
77
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
3.746947e-04
</td>
<td style="text-align:right;">
3.746947e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_ADIPOGENESIS
</td>
<td style="text-align:right;">
170
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
5.484190e-03
</td>
<td style="text-align:right;">
4.570158e-02
</td>
</tr>
</tbody>
</table>
<p>The barcode plots below show the enrichment of genes among the up and down regulated genes in the top four hallmark gene sets identified by the camera analysis. The black bars represent the logFC of each of the genes, with down regulated genes at the left and up regulated genes at the right of each plot. The enrichment worm at the top of the plot shows the relative enrichment of the genes in each part of the plot.</p>
<pre class="r"><code>par(mfrow=c(2,2))
sapply(1:4, function(i){
  barcodeplot(efit$t[,1], index=idx[[rownames(cam_h)[i]]], 
              main = rownames(cam_h)[i], cex.main = 0.75)
})</code></pre>
<p><img src="figure/FOX_limma.Rmd/hallmark_barcode-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL

[[4]]
NULL</code></pre>
<p>The curated gene sets are compiled from online pathway databases, publications in PubMed, and knowledge of domain experts.</p>
<pre class="r"><code>c2 &lt;- readRDS(&quot;./data/MSigDB/Hs.c2.all.v7.1.entrez.rds&quot;, refhook = NULL)
idx &lt;- ids2indices(c2, id=norm_FOX$genes$ENTREZID, remove.empty=TRUE)
cam_c2 &lt;- camera(vwts,idx,design1)
write.csv(cam_c2[cam_c2$FDR &lt; 0.05,], &quot;./output/camera_c2.csv&quot;, row.names = TRUE)
kable_styling(kable(cam_c2[cam_c2$FDR &lt; 0.05,][1:30,], format.args = list(scientific = -1), digits = 50, caption=&quot;Table 8: Top 30 Camera gene set test results for MSigDB curated gene sets&quot;))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Table 8: Top 30 Camera gene set test results for MSigDB curated gene sets
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
NGenes
</th>
<th style="text-align:left;">
Direction
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FLORIO_NEOCORTEX_BASAL_RADIAL_GLIA_DN
</td>
<td style="text-align:right;">
151
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
8.649180e-11
</td>
<td style="text-align:right;">
4.750995e-07
</td>
</tr>
<tr>
<td style="text-align:left;">
KEGG_SPLICEOSOME
</td>
<td style="text-align:right;">
125
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.042893e-08
</td>
<td style="text-align:right;">
2.864306e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_MRNA_SPLICING
</td>
<td style="text-align:right;">
185
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
5.023377e-08
</td>
<td style="text-align:right;">
9.197803e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
LEE_EARLY_T_LYMPHOCYTE_UP
</td>
<td style="text-align:right;">
89
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
8.988754e-08
</td>
<td style="text-align:right;">
1.216660e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
CHNG_MULTIPLE_MYELOMA_HYPERPLOID_UP
</td>
<td style="text-align:right;">
50
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.107464e-07
</td>
<td style="text-align:right;">
1.216660e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
KEGG_ECM_RECEPTOR_INTERACTION
</td>
<td style="text-align:right;">
50
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
1.936000e-07
</td>
<td style="text-align:right;">
1.561053e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
ZHAN_MULTIPLE_MYELOMA_PR_UP
</td>
<td style="text-align:right;">
37
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
1.989327e-07
</td>
<td style="text-align:right;">
1.561053e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_TRANSLATION
</td>
<td style="text-align:right;">
283
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
3.371653e-07
</td>
<td style="text-align:right;">
2.315061e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_17
</td>
<td style="text-align:right;">
176
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
5.253611e-07
</td>
<td style="text-align:right;">
3.000099e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
KANG_DOXORUBICIN_RESISTANCE_UP
</td>
<td style="text-align:right;">
48
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
5.835224e-07
</td>
<td style="text-align:right;">
3.000099e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
HSIAO_HOUSEKEEPING_GENES
</td>
<td style="text-align:right;">
382
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
6.007843e-07
</td>
<td style="text-align:right;">
3.000099e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_ECM_PROTEOGLYCANS
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
1.194170e-06
</td>
<td style="text-align:right;">
5.047693e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_AUF1_HNRNP_D0_BINDS_AND_DESTABILIZES_MRNA
</td>
<td style="text-align:right;">
53
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.194612e-06
</td>
<td style="text-align:right;">
5.047693e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
HOLLEMAN_PREDNISOLONE_RESISTANCE_B_ALL_UP
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.433215e-06
</td>
<td style="text-align:right;">
5.623323e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA
</td>
<td style="text-align:right;">
236
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
2.388443e-06
</td>
<td style="text-align:right;">
8.746479e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
DUTERTRE_ESTRADIOL_RESPONSE_24HR_UP
</td>
<td style="text-align:right;">
270
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
3.106083e-06
</td>
<td style="text-align:right;">
1.066357e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
VECCHI_GASTRIC_CANCER_ADVANCED_VS_EARLY_UP
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
3.459077e-06
</td>
<td style="text-align:right;">
1.117689e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_NEGATIVE_REGULATION_OF_NOTCH4_SIGNALING
</td>
<td style="text-align:right;">
54
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
3.993479e-06
</td>
<td style="text-align:right;">
1.218677e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_EUKARYOTIC_TRANSLATION_INITIATION
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
4.451415e-06
</td>
<td style="text-align:right;">
1.226245e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
MULLIGHAN_MLL_SIGNATURE_1_UP
</td>
<td style="text-align:right;">
358
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
4.464756e-06
</td>
<td style="text-align:right;">
1.226245e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_DECTIN_1_MEDIATED_NONCANONICAL_NF_KB_SIGNALING
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
5.726841e-06
</td>
<td style="text-align:right;">
1.497978e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
NAKAYAMA_SOFT_TISSUE_TUMORS_PCA2_UP
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
6.858841e-06
</td>
<td style="text-align:right;">
1.712528e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
KEGG_PARKINSONS_DISEASE
</td>
<td style="text-align:right;">
110
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
7.290392e-06
</td>
<td style="text-align:right;">
1.733851e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
KEGG_PROTEASOME
</td>
<td style="text-align:right;">
42
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
7.766479e-06
</td>
<td style="text-align:right;">
1.733851e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_MET_ACTIVATES_PTK2_SIGNALING
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
7.891185e-06
</td>
<td style="text-align:right;">
1.733851e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_METABOLISM_OF_POLYAMINES
</td>
<td style="text-align:right;">
56
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
9.421078e-06
</td>
<td style="text-align:right;">
1.940401e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
HONRADO_BREAST_CANCER_BRCA1_VS_BRCA2
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
9.537745e-06
</td>
<td style="text-align:right;">
1.940401e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
CREIGHTON_ENDOCRINE_THERAPY_RESISTANCE_2
</td>
<td style="text-align:right;">
303
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
9.951349e-06
</td>
<td style="text-align:right;">
1.952241e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_10
</td>
<td style="text-align:right;">
65
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.161878e-05
</td>
<td style="text-align:right;">
2.152445e-03
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_DEGRADATION_OF_GLI1_BY_THE_PROTEASOME
</td>
<td style="text-align:right;">
56
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.175557e-05
</td>
<td style="text-align:right;">
2.152445e-03
</td>
</tr>
</tbody>
</table>
<p>The barcode plots below show the top 4 curated genesets.</p>
<pre class="r"><code>par(mfrow=c(2,2))
sapply(1:4, function(i){
  barcodeplot(efit$t[,1], index=idx[[rownames(cam_c2)[i]]], 
              main = rownames(cam_c2)[i], cex.main = 0.75)
})</code></pre>
<p><img src="figure/FOX_limma.Rmd/c2_barcode-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL

[[4]]
NULL</code></pre>
</div>
<div id="summary" class="section level2">
<h2>Summary</h2>
<p>The sequences utilised in this study have previously been investigated using a similar analysis, which identified some overlapping significantly expressed genes. Some of the differences in the results can be attributed to the use of different methods for aligning and counting reads, and whether Ensembl or Entrez IDs were used. This new analysis, identified 9 genes that were not originally identified (<em>FN1</em>, <em>VSIG4</em>, <em>OLIG2</em>, <em>LOC105377267</em>, <em>RNVU1-2A</em>, <em>SMPD3</em>, <em>FOSB</em>, <em>TGM3</em> and <em>RPLP1</em>), as well as an additional two that were previously identified, but never validated (<em>CFAP54</em> and <em>CFAP45</em>). Additionally, this study used gene set enrichment analysis and competitive gene set testing to uncover pathways and gene sets that were perturbed in the blood of women with preterm growth restricted fetuses.</p>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] clusterProfiler_3.14.3                 
 [2] ggrepel_0.9.1                          
 [3] here_1.0.1                             
 [4] NMF_0.24.0                             
 [5] cluster_2.1.2                          
 [6] rngtools_1.5.2                         
 [7] pkgmaker_0.32.2                        
 [8] registry_0.5-1                         
 [9] patchwork_1.1.2                        
[10] Homo.sapiens_1.3.1                     
[11] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[12] org.Hs.eg.db_3.10.0                    
[13] GO.db_3.10.0                           
[14] OrganismDbi_1.28.0                     
[15] GenomicFeatures_1.38.2                 
[16] AnnotationDbi_1.48.0                   
[17] kableExtra_1.3.4                       
[18] knitr_1.40                             
[19] tableone_0.13.2                        
[20] RColorBrewer_1.1-3                     
[21] EDASeq_2.20.0                          
[22] ShortRead_1.44.3                       
[23] GenomicAlignments_1.22.1               
[24] SummarizedExperiment_1.16.1            
[25] DelayedArray_0.12.3                    
[26] matrixStats_0.62.0                     
[27] Rsamtools_2.2.3                        
[28] GenomicRanges_1.38.0                   
[29] GenomeInfoDb_1.22.1                    
[30] Biostrings_2.54.0                      
[31] XVector_0.26.0                         
[32] IRanges_2.20.2                         
[33] S4Vectors_0.24.4                       
[34] BiocParallel_1.20.1                    
[35] Biobase_2.46.0                         
[36] BiocGenerics_0.32.0                    
[37] forcats_0.5.2                          
[38] stringr_1.4.1                          
[39] dplyr_1.0.10                           
[40] purrr_0.3.4                            
[41] readr_2.1.2                            
[42] tidyr_1.2.1                            
[43] tibble_3.1.8                           
[44] ggplot2_3.3.6                          
[45] tidyverse_1.3.2                        
[46] edgeR_3.28.1                           
[47] Glimma_1.14.0                          
[48] limma_3.42.2                           
[49] workflowr_1.7.0                        

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3         rtracklayer_1.46.0     R.methodsS3_1.8.2     
  [4] bit64_4.0.5            aroma.light_3.16.0     R.utils_2.12.0        
  [7] data.table_1.14.2      hwriter_1.3.2.1        RCurl_1.98-1.8        
 [10] doParallel_1.0.17      generics_0.1.3         callr_3.7.2           
 [13] cowplot_1.1.1          RSQLite_2.2.17         europepmc_0.4.1       
 [16] proxy_0.4-27           bit_4.0.4              tzdb_0.3.0            
 [19] enrichplot_1.6.1       webshot_0.5.4          xml2_1.3.3            
 [22] lubridate_1.8.0        httpuv_1.6.6           assertthat_0.2.1      
 [25] viridis_0.6.2          gargle_1.2.1           xfun_0.33             
 [28] hms_1.1.2              jquerylib_0.1.4        evaluate_0.16         
 [31] promises_1.2.0.1       fansi_1.0.3            progress_1.2.2        
 [34] dbplyr_2.2.1           readxl_1.4.1           igraph_1.2.11         
 [37] DBI_1.1.3              geneplotter_1.64.0     googledrive_2.0.0     
 [40] ellipsis_0.3.2         backports_1.4.1        survey_4.1-1          
 [43] annotate_1.64.0        gridBase_0.4-7         biomaRt_2.42.1        
 [46] deldir_1.0-6           vctrs_0.4.1            cachem_1.0.6          
 [49] withr_2.5.0            ggforce_0.3.4          triebeard_0.3.0       
 [52] prettyunits_1.1.1      svglite_2.1.0          DOSE_3.12.0           
 [55] crayon_1.5.1           genefilter_1.68.0      labeling_0.4.2        
 [58] pkgconfig_2.0.3        tweenr_2.0.2           rlang_1.0.6           
 [61] lifecycle_1.0.2        BiocFileCache_1.10.2   modelr_0.1.9          
 [64] cellranger_1.1.0       rprojroot_2.0.3        polyclip_1.10-0       
 [67] graph_1.64.0           Matrix_1.3-4           urltools_1.7.3        
 [70] zoo_1.8-11             reprex_2.0.2           ggridges_0.5.4        
 [73] whisker_0.4            processx_3.7.0         googlesheets4_1.0.1   
 [76] png_0.1-7              viridisLite_0.4.1      bitops_1.0-7          
 [79] getPass_0.2-2          R.oo_1.25.0            blob_1.2.3            
 [82] qvalue_2.18.0          gridGraphics_0.5-1     jpeg_0.1-9            
 [85] scales_1.2.1           memoise_2.0.1          magrittr_2.0.3        
 [88] plyr_1.8.7             zlibbioc_1.32.0        compiler_3.6.1        
 [91] cli_3.4.1              ps_1.7.1               MASS_7.3-58.1         
 [94] tidyselect_1.1.2       stringi_1.7.8          highr_0.9             
 [97] mitools_2.4            yaml_2.3.5             GOSemSim_2.12.1       
[100] askpass_1.1            locfit_1.5-9.4         latticeExtra_0.6-30   
[103] grid_3.6.1             sass_0.4.2             fastmatch_1.1-3       
[106] tools_3.6.1            rstudioapi_0.14        foreach_1.5.2         
[109] git2r_0.30.1           gridExtra_2.3          farver_2.1.1          
[112] ggraph_2.0.6           rvcheck_0.2.1          digest_0.6.29         
[115] BiocManager_1.30.18    Rcpp_1.0.9             broom_1.0.1           
[118] later_1.3.0            httr_1.4.4             colorspace_2.0-3      
[121] rvest_1.0.3            XML_3.99-0.3           fs_1.5.2              
[124] splines_3.6.1          yulab.utils_0.0.5      RBGL_1.62.1           
[127] graphlayouts_0.8.1     ggplotify_0.1.0        systemfonts_1.0.4     
[130] xtable_1.8-4           jsonlite_1.8.0         tidygraph_1.2.2       
[133] R6_2.5.1               pillar_1.8.1           htmltools_0.5.3       
[136] glue_1.6.2             fastmap_1.1.0          DESeq_1.38.0          
[139] class_7.3-20           codetools_0.2-18       fgsea_1.12.0          
[142] utf8_1.2.2             lattice_0.20-45        bslib_0.4.0           
[145] curl_4.3.2             openssl_2.0.3          interp_1.0-33         
[148] survival_3.4-0         rmarkdown_2.16         munsell_0.5.0         
[151] e1071_1.7-11           DO.db_2.9              GenomeInfoDbData_1.2.2
[154] iterators_1.0.14       labelled_2.10.0        haven_2.5.1           
[157] reshape2_1.4.4         gtable_0.3.1          </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] clusterProfiler_3.14.3                 
 [2] ggrepel_0.9.1                          
 [3] here_1.0.1                             
 [4] NMF_0.24.0                             
 [5] cluster_2.1.2                          
 [6] rngtools_1.5.2                         
 [7] pkgmaker_0.32.2                        
 [8] registry_0.5-1                         
 [9] patchwork_1.1.2                        
[10] Homo.sapiens_1.3.1                     
[11] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[12] org.Hs.eg.db_3.10.0                    
[13] GO.db_3.10.0                           
[14] OrganismDbi_1.28.0                     
[15] GenomicFeatures_1.38.2                 
[16] AnnotationDbi_1.48.0                   
[17] kableExtra_1.3.4                       
[18] knitr_1.40                             
[19] tableone_0.13.2                        
[20] RColorBrewer_1.1-3                     
[21] EDASeq_2.20.0                          
[22] ShortRead_1.44.3                       
[23] GenomicAlignments_1.22.1               
[24] SummarizedExperiment_1.16.1            
[25] DelayedArray_0.12.3                    
[26] matrixStats_0.62.0                     
[27] Rsamtools_2.2.3                        
[28] GenomicRanges_1.38.0                   
[29] GenomeInfoDb_1.22.1                    
[30] Biostrings_2.54.0                      
[31] XVector_0.26.0                         
[32] IRanges_2.20.2                         
[33] S4Vectors_0.24.4                       
[34] BiocParallel_1.20.1                    
[35] Biobase_2.46.0                         
[36] BiocGenerics_0.32.0                    
[37] forcats_0.5.2                          
[38] stringr_1.4.1                          
[39] dplyr_1.0.10                           
[40] purrr_0.3.4                            
[41] readr_2.1.2                            
[42] tidyr_1.2.1                            
[43] tibble_3.1.8                           
[44] ggplot2_3.3.6                          
[45] tidyverse_1.3.2                        
[46] edgeR_3.28.1                           
[47] Glimma_1.14.0                          
[48] limma_3.42.2                           
[49] workflowr_1.7.0                        

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3         rtracklayer_1.46.0     R.methodsS3_1.8.2     
  [4] bit64_4.0.5            aroma.light_3.16.0     R.utils_2.12.0        
  [7] data.table_1.14.2      hwriter_1.3.2.1        RCurl_1.98-1.8        
 [10] doParallel_1.0.17      generics_0.1.3         callr_3.7.2           
 [13] cowplot_1.1.1          RSQLite_2.2.17         europepmc_0.4.1       
 [16] proxy_0.4-27           bit_4.0.4              tzdb_0.3.0            
 [19] enrichplot_1.6.1       webshot_0.5.4          xml2_1.3.3            
 [22] lubridate_1.8.0        httpuv_1.6.6           assertthat_0.2.1      
 [25] viridis_0.6.2          gargle_1.2.1           xfun_0.33             
 [28] hms_1.1.2              jquerylib_0.1.4        evaluate_0.16         
 [31] promises_1.2.0.1       fansi_1.0.3            progress_1.2.2        
 [34] dbplyr_2.2.1           readxl_1.4.1           igraph_1.2.11         
 [37] DBI_1.1.3              geneplotter_1.64.0     googledrive_2.0.0     
 [40] ellipsis_0.3.2         backports_1.4.1        survey_4.1-1          
 [43] annotate_1.64.0        gridBase_0.4-7         biomaRt_2.42.1        
 [46] deldir_1.0-6           vctrs_0.4.1            cachem_1.0.6          
 [49] withr_2.5.0            ggforce_0.3.4          triebeard_0.3.0       
 [52] prettyunits_1.1.1      svglite_2.1.0          DOSE_3.12.0           
 [55] crayon_1.5.1           genefilter_1.68.0      labeling_0.4.2        
 [58] pkgconfig_2.0.3        tweenr_2.0.2           rlang_1.0.6           
 [61] lifecycle_1.0.2        BiocFileCache_1.10.2   modelr_0.1.9          
 [64] cellranger_1.1.0       rprojroot_2.0.3        polyclip_1.10-0       
 [67] graph_1.64.0           Matrix_1.3-4           urltools_1.7.3        
 [70] zoo_1.8-11             reprex_2.0.2           ggridges_0.5.4        
 [73] whisker_0.4            processx_3.7.0         googlesheets4_1.0.1   
 [76] png_0.1-7              viridisLite_0.4.1      bitops_1.0-7          
 [79] getPass_0.2-2          R.oo_1.25.0            blob_1.2.3            
 [82] qvalue_2.18.0          gridGraphics_0.5-1     jpeg_0.1-9            
 [85] scales_1.2.1           memoise_2.0.1          magrittr_2.0.3        
 [88] plyr_1.8.7             zlibbioc_1.32.0        compiler_3.6.1        
 [91] cli_3.4.1              ps_1.7.1               MASS_7.3-58.1         
 [94] tidyselect_1.1.2       stringi_1.7.8          highr_0.9             
 [97] mitools_2.4            yaml_2.3.5             GOSemSim_2.12.1       
[100] askpass_1.1            locfit_1.5-9.4         latticeExtra_0.6-30   
[103] grid_3.6.1             sass_0.4.2             fastmatch_1.1-3       
[106] tools_3.6.1            rstudioapi_0.14        foreach_1.5.2         
[109] git2r_0.30.1           gridExtra_2.3          farver_2.1.1          
[112] ggraph_2.0.6           rvcheck_0.2.1          digest_0.6.29         
[115] BiocManager_1.30.18    Rcpp_1.0.9             broom_1.0.1           
[118] later_1.3.0            httr_1.4.4             colorspace_2.0-3      
[121] rvest_1.0.3            XML_3.99-0.3           fs_1.5.2              
[124] splines_3.6.1          yulab.utils_0.0.5      RBGL_1.62.1           
[127] graphlayouts_0.8.1     ggplotify_0.1.0        systemfonts_1.0.4     
[130] xtable_1.8-4           jsonlite_1.8.0         tidygraph_1.2.2       
[133] R6_2.5.1               pillar_1.8.1           htmltools_0.5.3       
[136] glue_1.6.2             fastmap_1.1.0          DESeq_1.38.0          
[139] class_7.3-20           codetools_0.2-18       fgsea_1.12.0          
[142] utf8_1.2.2             lattice_0.20-45        bslib_0.4.0           
[145] curl_4.3.2             openssl_2.0.3          interp_1.0-33         
[148] survival_3.4-0         rmarkdown_2.16         munsell_0.5.0         
[151] e1071_1.7-11           DO.db_2.9              GenomeInfoDbData_1.2.2
[154] iterators_1.0.14       labelled_2.10.0        haven_2.5.1           
[157] reshape2_1.4.4         gtable_0.3.1          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
