---
title: "FOX Case-Control limma-voom Differential Expression Analysis"
author: "Sally Beard"
date: "2022-09-27"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r, setup, include=FALSE}
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(warning = FALSE)
```

The sequence data used in the following analysis was generated at AGRF for the [FOX study](https://link.springer.com/article/10.1186/s12916-020-01605-x). RNA was isolated from whole blood PAXgene tubes and Illuminaâ€™s TruSeq stranded RNA sample preparation kit with globin depletion was used to prepare libraries for sequencing. Libraries were pooled and clustered using the Illumina cBot system, followed by sequencing on the Illumina HiSeq 2000 platform to obtain 50 base pair single-end reads. Each library was split over two lanes, which generated two sequence files per sample. Fastq files were aligned against the human genome GRCh38 using `hisat2`, then files were sorted with `SAMtools`, before reads aligning to exons were counted using `htseq` with default parameters. Technical duplicates run in different sequencing lanes overlapped on MDS plots, indicating there was no batch effect due to lane, therefore in this analysis counts from technical duplicates were summed for further analysis.

```{r}
#import required libraries
library(limma)
library(Glimma)
library(edgeR)
library(tidyverse)
library(EDASeq)
library(RColorBrewer)
library(tableone)
library(knitr)
library(kableExtra)
library(Homo.sapiens)
library(patchwork)
library(NMF)
library(here)
library(ggrepel)
library(clusterProfiler)
```

## Data import

Import sample information and create table of characteristics.

```{r}
metadata <- read.csv("data/FOX_REDCap_metadata.csv")
#create a variable list
vars <- c("Age", "BMI", "Diabetes", "Hypertension","GA_sampling", "GA_delivery", "Birthweight", "Birthweight_centile", "Sex", "Cord_blood_pH")

#create Table 1 stratified by case-control
tableOne <- CreateTableOne(vars = vars, strata = "Case", data = metadata, testNonNormal = kruskal.test)
kable_styling(kableone(tableOne, caption="Table 1: Patient characteristics for FOX cases included in RNA-seq analysis", nonnormal = c("Birthweight_centile", "Cord_blood_pH", "BMI"), varLabels = TRUE, dropEqual = TRUE))
```

Set up `DGElist` object for downstream analysis and add sample-specific information.

```{r}
#list count files from working directory
files = list.files("/Users/beards/rnaseq_FOX/FOX_case_control_counts", pattern = "[counts]$")

#edgeR has a function readDGE for combining separate count files into one matrix of counts in one step
counts <- readDGE(files, path = "/Users/beards/rnaseq_FOX/FOX_case_control_counts", columns = c(1,3), header = FALSE)

#remove meta tags as library size includes counts from the meta tags
MetaTags <- grep("^__", rownames(counts))
counts <- counts[-MetaTags, ]

#update library size to reflect mapped reads
counts$samples$lib.size <- colSums(counts$counts)

#obtain sample names from file names
sample <- strsplit2(files, "_")[,1]
counts$samples$sample <- as.factor(sample)
samplenames <- counts$samples$sample

#sum counts from technical replicates
counts <- sumTechReps(counts, ID=samplenames)

#Add sample-level information related to experimental design (and possible confounding factors) to the count matrix
counts$samples$site <- as.factor(metadata$Site)
counts$samples$sample <- as.factor(metadata$sample)
counts$samples$sex <- as.factor(metadata$Sex)
counts$samples$hypertension <- as.factor(metadata$Hypertension)
counts$samples$diabetes <- as.factor(metadata$Diabetes)
counts$samples$steroids <- as.factor(metadata$Antenatal_steroids)
counts$samples$group <- as.factor(metadata$Case)

#import information for each gene
geneid <- rownames(counts)
genes <- select(Homo.sapiens, keys=geneid, columns=c("SYMBOL", "TXCHROM", "ENTREZID"), 
                keytype="ENSEMBL")
genes <- genes[!duplicated(genes$ENSEMBL),]
counts$genes <- genes

#remove genes without ENTREZ IDs
keep <- !is.na(counts$genes$ENTREZID) & !is.null(counts$genes$ENTREZID)
counts <- counts[keep, ]

#generate library statistics
cpm <- cpm(counts)
lcpm <- cpm(counts, log=TRUE)
L <- mean(counts$samples$lib.size) * 1e-6
M <- median(counts$samples$lib.size) * 1e-6
```

## Quality Control

Genes with no or low counts are unlikely to be biologically important, they provide little evidence for differential expression, and interfere with statistical approximations, so they were filtered out before performing the analysis. Genes with low or no counts (counts per million reads that corresponded to less than ~10 reads in any 79 samples (smallest group size)) were filtered out of the analysis, based on the recommendations in the edgeR documentation. Both smaller (>2) and larger (<100) thresholds were tested, lower thresholds detected more significant genes on average, higher thresholds detected less differentially expressed genes, but those that were detected are likely to be more robust. The threshold chosen for the analysis is commonly used in the literature and provided a good compromise.

```{r}
#filter out genes with low expression
keep.exprs <- filterByExpr(counts, group=counts$samples$group)
counts_FOX <- counts[keep.exprs,, keep.lib.sizes=FALSE]
samplenames <- colnames(counts_FOX)
```

Plotting the distribution of log-CPM values (below) showed that before filtering (A) many genes in each sample had low or no expression, with negative log-CPM values. These genes were removed from the analysis after filtering (B). Filtering didn't completely improve the uniformity between samples however.

```{r denisty-plot}
#make density plots for counts before and after filtering out genes with low expression
par(mfrow = c(1,1))
lcpmz <- lcpm
lcpm.cutoff <- log2(10/M + 2/L)
nsamples <- ncol(counts)
col <- scales::hue_pal()(nsamples)
plot(density(lcpm[,1]), col=col[1], lwd=2, ylim=c(0,0.4), las=2, main="", xlab="")
  title(main="A. Raw data", xlab="Log-cpm")
abline(v=lcpm.cutoff, lty=3)
for (i in 2:nsamples){
  den <- density(lcpm[,i])
  lines(den$x, den$y, col=col[i], lwd=2)
}
lcpm2 <- cpm(counts_FOX, log=TRUE)
plot(density(lcpm2[,1]), col=col[1], lwd=2, ylim=c(0,0.4), las=2, main="", xlab="")
title(main="B. Filtered data", xlab="Log-cpm")
abline(v=lcpm.cutoff, lty=3)
for (i in 2:nsamples){
  den <- density(lcpm2[,i])
  lines(den$x, den$y, col=col[i], lwd=2)
}
```

We suspected that effective library size might have been contributing to the lack of uniformity between samples in the density plots. Colouring the samples by effective library size showed that the distribution of counts was similar for samples with larger libraries (8-12 million reads) where two peaks were seen after filtering, one at logCPM ~-1.5 and one at 5 (B). Samples with smaller libraries tended to have a single peak between ~3 and 5 log-CPM (B).

```{r density-plot-coloured-lib.size}
#create binds for effective samples size
lib.size <- counts_FOX$samples$lib.size
#set up cut-off values
breaks <- c(0,2000000,4000000,6000000,8000000,10000000,12000000)
#specify bin labels
tags <- c("[0-2m)", "[2-4m)", "[4-6m)", "[6-8m)", "[8-10m)", "[10-12m)")
#bucket library sizes into bins
group_tags <- cut(lib.size,
                  breaks=breaks,
                  include.lowest=TRUE,
                  right=FALSE,
                  labels=tags)
#add new tags to count matrix
counts_FOX$samples$lib_size <- as.factor(group_tags)

#make density plots for counts before and after filtering out genes with low expression
par(mfrow = c(1,1))
lcpmz <- lcpm
lcpm.cutoff <- log2(10/M + 2/L)
nsamples <- ncol(counts)
col.lib <- counts_FOX$samples$lib_size
levels(col.lib) <- brewer.pal(nlevels(col.lib), "Set1")
plot(density(lcpmz[,1]), col = as.character(col.lib[1]), lwd = 1, ylim = c(0, 0.4), las = 2, 
     main = "", xlab = "")
title(main = "A. Raw data", xlab = "Log-cpm")
abline(v = lcpm.cutoff, lty = 3)
for (i in 2:nsamples){
  den <- density(lcpmz[,i])
  lines(den$x, den$y, col = as.character(col.lib[i]), lwd = 1)
}
legend("topright",legend=levels(counts_FOX$samples$lib_size), col=levels(col.lib), pch=16, title ="Library size (millions)")
lcpmy <- cpm(counts_FOX, log=TRUE)
plot(density(lcpmy[,1]), col = as.character(col.lib[1]), lwd = 1, ylim = c(0, 0.4), las = 2, 
     main = "", xlab = "")
title(main = "B. Filtered data", xlab = "Log-cpm")
abline(v = lcpm.cutoff, lty = 3)
for (i in 2:nsamples){
  den <- density(lcpmy[,i])
  lines(den$x, den$y, col = as.character(col.lib[i]), lwd = 1)
}
legend("topright",legend=levels(counts_FOX$samples$lib_size), col=levels(col.lib), pch=16, title ="Library size (millions)" )
```

TMM normalisation was applied to account for any highly expressed genes monopolising the sequencing and to account for the underlying differences in the distributions of expressed transcripts between samples.

```{r}
#apply TMM normalisation
norm_FOX <- calcNormFactors(counts_FOX, method = "TMM")
```

Although ~30 million reads per sample were obtained from the sequencing, effective library sizes varied from around 2 million reads to more than 10 million reads, with the median just under 7 million reads (A). This suggests we are only likely to capture the most abundant maternal RNAs. The range and distribution of expression values look fairly uniform between samples,  which is what we expect, and TMM normalisation further improves this.

```{r expression-plot}
dat <- data.frame(lib = norm_FOX$samples$lib.size,
                  status = norm_FOX$samples$group,
                  sample = colnames(norm_FOX))
p1 <- ggplot(dat, aes(x = sample, y = lib, fill = status)) +
  geom_bar(stat = "identity") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = "Sample", y = "Library size",
       fill = "Preeclampsia status", title = "A. Library size after filtering") +
  geom_hline(yintercept = median(dat$lib), linetype = "dashed") +
  scale_x_discrete(limits = dat$sample)

dat <- reshape2::melt(cpm(norm_FOX, normalized.lib.sizes = FALSE, log = TRUE),
                      value.name = "cpm")
dat$status <- rep(norm_FOX$samples$group, each = nrow(norm_FOX))
colnames(dat)[2] <- "sample"
p2 <- ggplot(dat, aes(x = sample, y = cpm, fill = status)) +
  geom_boxplot(show.legend = FALSE, outlier.size = 0.75) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7)) +
  labs(x = "Sample", y = "Log-CPM",
       fill = "Preeclampsia status", title = "B. Expression values after filtering") +
  geom_hline(yintercept = median(dat$lib), linetype = "dashed")

dat <- reshape2::melt(cpm(norm_FOX, normalized.lib.sizes = TRUE, log = TRUE),
                      value.name = "cpm")

dat$status <- rep(norm_FOX$samples$group, each = nrow(norm_FOX))
colnames(dat)[2] <- "sample"
p3 <- ggplot(dat, aes(x = sample, y = cpm, fill = status)) +
  geom_boxplot(show.legend = FALSE, outlier.size = 0.75) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7)) +
  labs(x = "Sample", y = "Log-CPM",
       fill = "Status", title = "C. Expression values after normalisation") +
  geom_hline(yintercept = median(dat$lib), linetype = "dashed")

p1
p2
p3
```

Multi-dimensional scaling (MDS) plots show the largest sources of variation in the data. They are a good way of identifying structure in the data and exploring relationships between samples. The following plots examine the first four principle components coloured for known features of the data including preeclampsia status, fetal sex, collection site, diabates status, steroid treatment prior to blood collection and library size. The samples did not cluster by known features of the data, except library size. The lack of clustering by case-control status in any of the first four principle components suggests we might not find any differentially expressed genes. The MDS plots suggest that the largest source of variation in the data is library size, as there is a strong trend towards increasingly smaller library sizes from left to right across the first dimension.

```{r mds-plot}
dims <- list(c(1,2), c(1,3), c(2,3), c(3,4))
vars <- c("group", "sex", "site", "diabetes", "steroids", "lib_size")
patches <- vector("list", length(vars))

for(i in 1:length(vars)){
  p <- vector("list", length(dims))
  
  for(j in 1:length(dims)){
    mds <- plotMDS(cpm(norm_FOX, log = TRUE), top = 1000, gene.selection="common", 
                   plot = FALSE, dim.plot = dims[[j]])
    dat <- tibble::tibble(x = mds$x, y = mds$y,
                          sample = samplenames,
                          variable = pull(norm_FOX$samples, vars[i]))
    
    p[[j]] <- ggplot(dat, aes(x = x, y = y, colour = variable)) +
      geom_text(aes(label = sample), size = 2.5) +
      labs(x = glue::glue("Principal component {dims[[j]][1]}"), 
           y = glue::glue("Principal component {dims[[j]][2]}"),
           colour = vars[i])
  }
  
  patches[[i]] <- wrap_elements(wrap_plots(p, ncol = 2, guides = "collect") +
    plot_annotation(title = glue::glue("Coloured by: {vars[i]}")) &
    theme(legend.position = "bottom"))
  
}

patches
```

## Differential expression analysis

Due to the variability in the data and the strong influence of library size, the TMM normalised data was transformed using `voomWithQualityWeights`. This takes into account the different library sizes and the mean variance relationship in the data and calculates sample-specific quality weights. Linear models were fit in `limma`, incorporating the `voom` weights. The samples from preterm growth restricted pregnancies were compared to controls. A summary of the number of differentially expressed genes (table 2) and their details (table 3) are shown below.

```{r voom-output}
#relevel the groups so that control is the reference
group <- as.factor(norm_FOX$samples$group)
group <- relevel(group, "Control")

#set up design matrix to test case status
design1 <- model.matrix(~group)
colnames(design1) <- gsub("group", "", colnames(design1))
rownames(design1) <- samplenames

vwts <- voomWithQualityWeights(norm_FOX, design1, plot=TRUE)
vfit <- lmFit(vwts, design1)
vfit <- contrasts.fit(vfit, coef = 2)
efit <- eBayes(vfit)

kable_styling(kable(summary(decideTests(efit)), caption="Table 2: Number of differentially expressed genes"))

top_table <- topTable(efit, n=40, sort.by = "P")
write.csv(topTable(efit, n=Inf), "./output/voomQual.csv")
kable_styling(kable(top_table[,c(3,5,6,9)], caption="Table 3: Top 40 DE genes in cases compared to controls", format.args = list(scientific = -1), digits = 50))
```

Since so many DE genes were identified, a logFC cut-off was applied using the [`treat`](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2654802/) method, which can be used to calculate p-values from empirical Bayes moderated t-statistics with a minimum log-FC requirement. The cutoff was set to 0.5, which is means that only genes that have at least a 1.4-fold change in expression in cases compared to controls will be detected. A summary of the number of differentially expressed genes (table 3) and their details (table 4) are shown below.

```{r}
tfit <- treat(vfit, lfc=0.5)
kable_styling(kable(summary(decideTests(tfit)), caption="Table 3: Number of differentially expressed genes from treat method"))

top_treat <- data.frame(topTreat(tfit, n=20, sort.by = "P"))
write.csv(top_treat, "./output/voomQual_treat.csv")
kable_styling(kable(top_treat[,c(3,5,6,9)], caption="Table 4: DE genes in cases compared to controls with treat method", format.args = list(scientific = -1), digits = 50))
```

The volcano plot below shows the statistical significance versus the logFC of genes in the differential expression analysis with the genes identified using the treat method represented in red.

```{r volcano-plot}
topTreat(tfit, num = Inf) %>% 
  mutate(sig = ifelse(adj.P.Val <= 0.05, "<= 0.05", "> 0.05")) -> dat

ggplot(dat, aes(x = logFC, y = -log10(P.Value), color = sig)) +
  geom_point(alpha = 0.75) +
  ggrepel::geom_text_repel(data = subset(dat, adj.P.Val < 0.05), 
            aes(x = logFC, y = -log10(P.Value), 
                label = SYMBOL), 
            size = 2, colour = "black", max.overlaps = 15) +
  labs(x = expression(~log[2]~"(Fold Change)"), 
       y = expression(~-log[10]~"(P-value)"),
       colour = "FDR") +
  scale_colour_brewer(palette = "Set1")
```

The following plots show the expression of the top 12 differentially expressed genes for samples collected from cases compared to controls. Despite the variability in the data and the lack of separation in the MDS plots, we were still able to detect differentially expressed genes, with varying degrees of overlap in expression.

```{r gene-expression}
dat <- reshape2::melt(cpm(norm_FOX, log = TRUE),
                      value.name = "cpm")
dat$group <- rep(norm_FOX$samples$group, each = nrow(norm_FOX))
dat$group <- relevel(dat$group, "Control")
dat$gene <- rep(norm_FOX$genes$SYMBOL, ncol(norm_FOX))

p <- vector("list", 12)

for(i in 1:length(p)){
  p[[i]] <-ggplot(data = subset(dat, dat$gene == top_treat$SYMBOL[i]), 
       aes(x = group, y = cpm, colour = group)) +
  geom_jitter(width = 0.25) +
  stat_summary(fun = "mean", geom = "crossbar") +
  labs(x = "Status", y = "log2 Norm. CPM", colour = "Status") +
  ggtitle(top_treat$SYMBOL[i]) +
  theme(plot.title = element_text(size = 8),
        plot.subtitle = element_text(size = 7),
        axis.title = element_text(size = 8),
        axis.text.x = element_text(size = 7))
}

wrap_plots(p, guides = "collect", ncol = 4) & 
  theme(legend.position = "bottom")
```

## Gene set enrichment analysis

Testing for enrichment of Gene Ontology (GO) categories among statistically significant differentially expressed genes (from treat analysis) using `goana` didn't yield any significant GO categories after adjusting for multiple testing.

```{r}
go <- goana(top_treat$ENTREZID[top_treat$adj.P.Val < 0.05], universe = norm_FOX$genes$entrezid)
goana_treat <- topGO(go, number = Inf) %>%
  mutate(FDR = p.adjust(P.DE)) %>%
  dplyr::filter(FDR < 0.05)
write.csv(goana_treat, "./output/goana_treat.csv", row.names = TRUE)
kable_styling(knitr::kable(goana_treat, format.args = list(scientific = -1), digits = 50))
```

Since GO categories overlap, the approach outlined above may be too stringent, so we also used the over-representation test implemented in [`clusterProfiler`](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3339379/) among the genes identified in the treat analysis, but only two categories were identified (below).

```{r}
gene <- top_treat$ENTREZID
universe <- norm_FOX$genes$ENTREZID
ego <- enrichGO(gene          = gene,
                universe      = universe,
                OrgDb         = org.Hs.eg.db,
                ont           = "ALL",
                pAdjustMethod = "BH",
                qvalueCutoff  = 0.05,
        readable      = TRUE)
dotplot(ego)
```

When we tested for enrichment of GO categories among all statistically significant genes (regardless of logFC) using goana, many more categories were identified (tbale 5), but they were not very specific.

```{r}
top_table <- data.frame(topTable(efit, number = Inf, sort.by = "P"))
go <- goana(top_table$ENTREZID[top_table$adj.P.Val < 0.05], universe = norm_FOX$genes$entrezid)
go_all <- topGO(go, number = Inf) %>%
  mutate(FDR = p.adjust(P.DE)) %>%
  dplyr::filter(FDR < 0.05)
write.csv(go_all, "./output/goana_all.csv", row.names = TRUE)
kable_styling(knitr::kable(go_all[1:40,], format.args = list(scientific = -1), digits = 50, caption="Table 5: Top 40 GO categories from goana gene set enrichment analysis using all genes with FDR < 0.05"))
```

Using `clusterProfiler` with this larger set of genes as input yielded 10 significant GO categories related to neutrophils (below). 

```{r}
gene <- top_table$ENTREZID[top_table$adj.P.Val < 0.05]
universe <- norm_FOX$genes$ENTREZID
ego <- enrichGO(gene          = gene,
                universe      = universe,
                OrgDb         = org.Hs.eg.db,
                ont           = "ALL",
                pAdjustMethod = "BH",
                qvalueCutoff  = 0.05,
        readable      = TRUE)
dotplot(ego)
```

The `camera` function performs a competitive test to find if genes in a given set are highly ranked in terms of differential expression compared to genes that are not in that set. In the output below we tested gene sets from the Molecular Signatures Database ([MSigDB](https://www.gsea-msigdb.org/gsea/msigdb/index.jsp_)).

MSigDB genesets from version 7.1 in R format were downloaded [here](https://bioinf.wehi.edu.au/MSigDB/v7.1/) for this analysis. We looked at GO gene sets, Hallmark gene sets and curated gene sets.

The GO gene sets consist of genes annotated by the same GO terms.

```{r}
GO <- readRDS("./data/MSigDB/Hs.c5.all.v7.1.entrez.rds", refhook = NULL)
idx <- ids2indices(GO, id=norm_FOX$genes$ENTREZID, remove.empty=TRUE)
cam_GO <- camera(vwts,idx,design1)
write.csv(cam_GO[cam_GO$FDR < 0.05,], "./output/camera_GO.csv", row.names = TRUE)
kable_styling(kable(cam_GO[cam_GO$FDR < 0.05,][1:30,], format.args = list(scientific = -1), digits = 50, caption="Table 6: Camera gene set test results for MSigDB GO gene sets"))
```

The Hallmark gene sets are coherently expressed signatures derived by aggregating many MSigDB gene sets to represent well-defined biological states or processes.

```{r}
hallmark <- readRDS("./data/MSigDB/Hs.h.all.v7.1.entrez.rds", refhook = NULL)
idx <- ids2indices(hallmark, id=norm_FOX$genes$ENTREZID, remove.empty=TRUE)
cam_h <- camera(vwts,idx,design1)
write.csv(cam_h[cam_h$FDR < 0.05,], "./output/camera_hallmark.csv", row.names = TRUE)
kable_styling(kable(cam_h[cam_h$FDR < 0.05,], format.args = list(scientific = -1), digits = 50, caption="Table 7: Camera gene set test results for MSigDB Hallmark gene sets"))
```

The barcode plots below show the enrichment of genes among the up and down regulated genes in the top four hallmark gene sets identified by the camera analysis. The black bars represent the logFC of each of the genes, with down regulated genes at the left and up regulated genes at the right of each plot. The enrichment worm at the top of the plot shows the relative enrichment of the genes in each part of the plot.

```{r hallmark_barcode}
par(mfrow=c(2,2))
sapply(1:4, function(i){
  barcodeplot(efit$t[,1], index=idx[[rownames(cam_h)[i]]], 
              main = rownames(cam_h)[i], cex.main = 0.75)
})
```

The curated gene sets are compiled from online pathway databases, publications in PubMed, and knowledge of domain experts. 

```{r}
c2 <- readRDS("./data/MSigDB/Hs.c2.all.v7.1.entrez.rds", refhook = NULL)
idx <- ids2indices(c2, id=norm_FOX$genes$ENTREZID, remove.empty=TRUE)
cam_c2 <- camera(vwts,idx,design1)
write.csv(cam_c2[cam_c2$FDR < 0.05,], "./output/camera_c2.csv", row.names = TRUE)
kable_styling(kable(cam_c2[cam_c2$FDR < 0.05,][1:30,], format.args = list(scientific = -1), digits = 50, caption="Table 8: Top 30 Camera gene set test results for MSigDB curated gene sets"))
```

The barcode plots below show the top 4 curated genesets.

```{r c2_barcode}
par(mfrow=c(2,2))
sapply(1:4, function(i){
  barcodeplot(efit$t[,1], index=idx[[rownames(cam_c2)[i]]], 
              main = rownames(cam_c2)[i], cex.main = 0.75)
})
```

## Summary

The sequences utilised in this study have previously been investigated using a similar analysis, which identified some overlapping significantly expressed genes. Some of the differences in the results can be attributed to the use of different methods for aligning and counting reads, and whether Ensembl or Entrez IDs were used. This new analysis, identified 9 genes that were not originally identified (*FN1*, *VSIG4*, *OLIG2*, *LOC105377267*, *RNVU1-2A*, *SMPD3*, *FOSB*, *TGM3* and *RPLP1*), as well as an additional two that were previously identified, but never validated (*CFAP54* and *CFAP45*). Additionally, this study used gene set enrichment analysis and competitive gene set testing to uncover pathways and gene sets that were perturbed in the blood of women with preterm growth restricted fetuses.

```{r}
sessionInfo()
```

